<?xml version="1.0" encoding="UTF-8"?><rss version="2.0"
	xmlns:content="http://purl.org/rss/1.0/modules/content/"
	xmlns:wfw="http://wellformedweb.org/CommentAPI/"
	xmlns:dc="http://purl.org/dc/elements/1.1/"
	xmlns:atom="http://www.w3.org/2005/Atom"
	xmlns:sy="http://purl.org/rss/1.0/modules/syndication/"
	xmlns:slash="http://purl.org/rss/1.0/modules/slash/"
	>

<channel>
	<title>Seqan</title>
	<atom:link href="https://www.seqan.de/feed/" rel="self" type="application/rss+xml" />
	<link>https://www.seqan.de</link>
	<description></description>
	<lastBuildDate>Tue, 13 Oct 2020 11:29:01 +0000</lastBuildDate>
	<language>en-US</language>
	<sy:updatePeriod>
	hourly	</sy:updatePeriod>
	<sy:updateFrequency>
	1	</sy:updateFrequency>
	<generator>https://wordpress.org/?v=5.6.1</generator>

<image>
	<url>https://www.seqan.de/wp-content/uploads/2020/11/cropped-seqan_logo_4-2-32x32.png</url>
	<title>Seqan</title>
	<link>https://www.seqan.de</link>
	<width>32</width>
	<height>32</height>
</image> 
	<item>
		<title>SeqAn 3.0.2 released</title>
		<link>https://www.seqan.de/seqan-3-0-2-release/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Tue, 13 Oct 2020 11:23:10 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=13313</guid>

					<description><![CDATA[Dear SeqAn users, supporters, and subscribers, today we present a new update of the SeqAn library. Hooray!!! The following highlights are present in the current release. General C++-20 support The C++-20 support for concepts and type-traits has been fully integrated into SeqAn. This means that even in C++-17 mode you can use the concept [...]]]></description>
										<content:encoded><![CDATA[<div class="fusion-fullwidth fullwidth-box fusion-builder-row-1 nonhundred-percent-fullwidth non-hundred-percent-height-scrolling"  style='background-color: rgba(255,255,255,0);background-position: center center;background-repeat: no-repeat;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px;'><div class="fusion-builder-row fusion-row "><div  class="fusion-layout-column fusion_builder_column fusion_builder_column_1_1 fusion-builder-column-0 fusion-one-full fusion-column-first fusion-column-last 1_1"  style='margin-top:0px;margin-bottom:0px;'><div class="fusion-column-wrapper" style="padding: 0px 0px 0px 0px;background-position:left top;background-repeat:no-repeat;-webkit-background-size:cover;-moz-background-size:cover;-o-background-size:cover;background-size:cover;"   data-bg-url=""><div class="fusion-text"></p>
<div id="magicdomid5" class=""><span class="author-g-eb7li037lghr9j81">Dear SeqAn users, supporters, and subscribers,</span></div>
<div id="magicdomid6" class=""></div>
<div id="magicdomid7" class=""><span class="author-g-eb7li037lghr9j81">today we present a new update of the SeqAn library. Hooray!!!</span></div>
<div id="magicdomid177" class="ace-line"><span class="author-g-eb7li037lghr9j81">The following highlights are present in the current release.</span></div>
<h2 id="magicdomid180" class="ace-line"><span class="author-g-eb7li037lghr9j81">General C++-20 support</span></h2>
<div id="magicdomid182" class="ace-line"><span class="author-g-eb7li037lghr9j81">The </span><span class="author-g-ojf51wntp1sx7fao">C</span><span class="author-g-eb7li037lghr9j81">++-20 support for concepts and type-traits has been fully integrated into SeqAn.</span></div>
<div id="magicdomid183" class="ace-line"><span class="author-g-eb7li037lghr9j81">This means that even in </span><span class="author-g-ojf51wntp1sx7fao">C</span><span class="author-g-eb7li037lghr9j81">++-17 mode you can use the concept names and type traits as defined by the </span><span class="author-g-ojf51wntp1sx7fao">C</span><span class="author-g-eb7li037lghr9j81">++20-specification</span></div>
<div id="magicdomid184" class="ace-line"><span class="author-g-eb7li037lghr9j81">and switch to </span><span class="author-g-ojf51wntp1sx7fao">C</span><span class="author-g-eb7li037lghr9j81">++-20 without breaking your code.</span></div>
<h2 id="magicdomid186" class="ace-line"><span class="author-g-eb7li037lghr9j81">Cookbook</span></h2>
<div id="magicdomid188" class="ace-line"><span class="author-g-eb7li037lghr9j81">We present the new <a href="https://docs.seqan.de/seqan/3.0.2/cookbook.html">Cookbook</a></span><span class="author-g-eb7li037lghr9j81">, where you can find useful </span></div>
<div id="magicdomid189" class="ace-line"><span class="author-g-eb7li037lghr9j81">code examples for typical tasks with SeqAn, e.g. reading a sequence file, or searching for a pattern in a string. </span></div>
<h2 id="magicdomid191" class="ace-line"><span class="author-g-eb7li037lghr9j81">Application template</span></h2>
<div id="magicdomid193" class="ace-line"><span class="author-g-eb7li037lghr9j81">In order to get started with new SeqAn3 applications easily, we introduce the new </span></div>
<div id="magicdomid194" class="ace-line"><span class="author-g-eb7li037lghr9j81"><a href="https://github.com/seqan/app-template">Application Template</a></span><span class="author-g-eb7li037lghr9j81">, where you can check out a template application with </span></div>
<div id="magicdomid195" class="ace-line"><span class="author-g-eb7li037lghr9j81">build system, tests and all necessary requirements set up. </span></div>
<h2 id="magicdomid11"><span class="author-g-eb7li037lghr9j81">User callback in alignment and search</span></h2>
<div id="magicdomid13" class=""><span class="author-g-eb7li037lghr9j81">The pairwise alignment can now be configured with a user-defined callback, which is called for every computed alignment</span></div>
<div id="magicdomid14" class=""><span class="author-g-eb7li037lghr9j81">result instead of returning a lazy range over the alignment results:</span></div>
<div></div>
<div><style type="text/css" scopped="scopped">.fusion-syntax-highlighter-1 > .CodeMirror .CodeMirror-gutters { background-color: ; }.fusion-syntax-highlighter-1 > .CodeMirror .CodeMirror-linenumber { color: ; }</style><div class="fusion-syntax-highlighter-container fusion-syntax-highlighter-1 fusion-syntax-highlighter-theme-dark" style="opacity:0;margin-top:0px;margin-right:0px;margin-bottom:0px;margin-left:0px;font-size:14px;border-width:1px;border-style:solid;border-color:#e0dede;"><div class="syntax-highlighter-copy-code"><span class="syntax-highlighter-copy-code-title" data-id="fusion_syntax_highlighter_1" style="font-size:14px;">Copy to Clipboard</span></div><textarea class="fusion-syntax-highlighter-textarea" id="fusion_syntax_highlighter_1" data-readOnly="nocursor" data-lineNumbers="1" data-lineWrapping="" data-theme="oceanic-next">#include <mutex>
#include <vector>

#include <seqan3/alignment/configuration/align_config_edit.hpp>
#include <seqan3/alignment/configuration/align_config_method.hpp>
#include <seqan3/alignment/configuration/align_config_on_result.hpp>
#include <seqan3/alignment/configuration/align_config_parallel.hpp>
#include <seqan3/alignment/pairwise/align_pairwise.hpp>
#include <seqan3/alphabet/nucleotide/dna4.hpp>
#include <seqan3/core/debug_stream.hpp>

int main()
{
    // Generate some sequences.
    using seqan3::operator""_dna4;
    using sequence_pair_t = std::pair<seqan3::dna4_vector, seqan3::dna4_vector>;
    std::vector<sequence_pair_t> sequences{100, {"AGTGCTACG"_dna4, "ACGTGCGACTAG"_dna4}};

    std::mutex write_to_debug_stream{}; // Need mutex to synchronise the output.

    // Use edit distance with 4 threads.
    auto const alignment_config = seqan3::align_cfg::method_global{} |
                                  seqan3::align_cfg::edit_scheme |
                                  seqan3::align_cfg::parallel{4} |
                                  seqan3::align_cfg::on_result{[&] (auto && result)
                                  {
                                      std::lock_guard sync{write_to_debug_stream}; // critical section
                                      seqan3::debug_stream << result << '\n';
                                  }};

    // Compute the alignments in parallel, and output them unordered using the callback (order is not deterministic).
    seqan3::align_pairwise(sequences, alignment_config);  // seqan3::align_pairwise is now declared void.
}</textarea></div></div>
<div id="magicdomid52" class=""></div>
<div id="magicdomid196" class="ace-line"><span class="author-g-eb7li037lghr9j81">This feature was also added to the `seqan3::search` algorithm, i.e. both algorithms can be customised using a user callback.</span></div>
<h2 id="magicdomid56"><span class="author-g-eb7li037lghr9j81">Lazy search result range</span></h2>
<div id="magicdomid58" class=""><span class="author-g-eb7li037lghr9j81">The return type of the search algorithm was adapted to use a lazy result range over the found hits during the </span></div>
<div id="magicdomid59" class=""><span class="author-g-eb7li037lghr9j81">search. This allows you to use the same code to step through the hits independently of having an index over a single</span></div>
<div id="magicdomid60" class=""><span class="author-g-eb7li037lghr9j81">text or a text collection:</span></div>
<div id="magicdomid61" class=""></div>
<div><style type="text/css" scopped="scopped">.fusion-syntax-highlighter-2 > .CodeMirror .CodeMirror-gutters { background-color: ; }.fusion-syntax-highlighter-2 > .CodeMirror .CodeMirror-linenumber { color: ; }</style><div class="fusion-syntax-highlighter-container fusion-syntax-highlighter-2 fusion-syntax-highlighter-theme-dark" style="opacity:0;margin-top:0px;margin-right:0px;margin-bottom:0px;margin-left:0px;font-size:14px;border-width:1px;border-style:solid;border-color:#e0dede;"><div class="syntax-highlighter-copy-code"><span class="syntax-highlighter-copy-code-title" data-id="fusion_syntax_highlighter_2" style="font-size:14px;">Copy to Clipboard</span></div><textarea class="fusion-syntax-highlighter-textarea" id="fusion_syntax_highlighter_2" data-readOnly="nocursor" data-lineNumbers="1" data-lineWrapping="" data-theme="oceanic-next">#include <vector>

#include <seqan3/alphabet/nucleotide/dna4.hpp>
#include <seqan3/core/debug_stream.hpp>
#include <seqan3/search/configuration/max_error.hpp>
#include <seqan3/search/configuration/hit.hpp>
#include <seqan3/search/search.hpp>
#include <seqan3/search/fm_index/fm_index.hpp>

int main()
{
    using seqan3::operator""_dna4;

    std::vector<seqan3::dna4_vector> text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTA"_dna4,
                                          "ACCCGATGAGCTACCCAGTAGTCGAACTG"_dna4,
                                          "GGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
    seqan3::dna4_vector query{"GCT"_dna4};

    seqan3::configuration const search_config = seqan3::search_cfg::max_error_total{seqan3::search_cfg::error_count{1}} |
                                                seqan3::search_cfg::hit_all_best{};

    // Always provide a unified interface over the found hits independent of the index its text layout.
    seqan3::debug_stream << "Search in text collection:\n";
    seqan3::fm_index index_collection{text};
    for (auto && hit : search(query, index_collection, search_config)) // Over a text collection.
        seqan3::debug_stream << hit << '\n';

    seqan3::debug_stream << "\nSearch in single text:\n";
    seqan3::fm_index index_single{text[0]};
    for (auto && hit : search(query, index_single, search_config)) // Over a single text.
        seqan3::debug_stream << hit << '\n';
}
</textarea></div></div>
<h2 id="magicdomid97"><span class="author-g-eb7li037lghr9j81">Dynamic hit configuration</span></h2>
<div id="magicdomid99" class=""><span class="author-g-eb7li037lghr9j81">The hit configuration can now be set dynamically:</span></div>
<div id="magicdomid100" class=""></div>
<div><style type="text/css" scopped="scopped">.fusion-syntax-highlighter-3 > .CodeMirror .CodeMirror-gutters { background-color: ; }.fusion-syntax-highlighter-3 > .CodeMirror .CodeMirror-linenumber { color: ; }</style><div class="fusion-syntax-highlighter-container fusion-syntax-highlighter-3 fusion-syntax-highlighter-theme-dark" style="opacity:0;margin-top:0px;margin-right:0px;margin-bottom:0px;margin-left:0px;font-size:14px;border-width:1px;border-style:solid;border-color:#e0dede;"><div class="syntax-highlighter-copy-code"><span class="syntax-highlighter-copy-code-title" data-id="fusion_syntax_highlighter_3" style="font-size:14px;">Copy to Clipboard</span></div><textarea class="fusion-syntax-highlighter-textarea" id="fusion_syntax_highlighter_3" data-readOnly="nocursor" data-lineNumbers="1" data-lineWrapping="" data-theme="oceanic-next">#include <vector>

#include <seqan3/alphabet/nucleotide/dna4.hpp>
#include <seqan3/core/debug_stream.hpp>
#include <seqan3/search/configuration/max_error.hpp>
#include <seqan3/search/configuration/hit.hpp>
#include <seqan3/search/fm_index/fm_index.hpp>
#include <seqan3/search/search.hpp>

int main()
{
    using seqan3::operator""_dna4;

    std::vector<seqan3::dna4_vector> text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTA"_dna4,
                                          "ACCCGATGAGCTACCCAGTAGTCGAACTG"_dna4,
                                          "GGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
    seqan3::dna4_vector query{"GCT"_dna4};
    seqan3::fm_index index{text};

    // Use the dynamic hit configuration to set hit_all_best mode.
    seqan3::configuration search_config = seqan3::search_cfg::max_error_total{seqan3::search_cfg::error_count{1}} |
                                          seqan3::search_cfg::hit{seqan3::search_cfg::hit_all_best{}};

    seqan3::debug_stream << "All single best hits:\n";
    for (auto && hit : search(query, index, search_config)) // Find all best hits:
        seqan3::debug_stream << hit << '\n';

    // Change the hit configuration to the strata mode with a stratum of 1.
    using seqan3::get;
    get<seqan3::search_cfg::hit>(search_config).hit_variant = seqan3::search_cfg::hit_strata{1};

    seqan3::debug_stream << "\nAll x+1 hits:\n";
    for (auto && hit : search(query, index, search_config)) // Find all strata hits.
        seqan3::debug_stream << hit << '\n';
}</textarea></div></div>
<h2 id="magicdomid156"><span class="author-g-eb7li037lghr9j81">Other important changes and additions</span></h2>
<div id="magicdomid158" class=""><span class="author-g-eb7li037lghr9j81">We put much effort into harmonising the algorithm configuration interface for the pairwise alignment and </span></div>
<div id="magicdomid159" class=""><span class="author-g-eb7li037lghr9j81">search algorithm. This involves some structural changes, which break the current code, but make the interfaces much more </span></div>
<div id="magicdomid160" class=""><span class="author-g-eb7li037lghr9j81">stable and easier to use.</span></div>
<div id="magicdomid203" class="ace-line"><span class="author-g-eb7li037lghr9j81">SeqAn</span><span class="author-g-l4p5hayyb3txbm6e"> n</span><span class="author-g-d4phsgoja0lb2x42">ow</span><span class="author-g-eb7li037lghr9j81"> offers</span> <span class="author-g-eb7li037lghr9j81">a data structure called interleaved Bloom filter, which can answer set-membership queries efficiently.</span></div>
<div id="magicdomid162" class=""><span class="author-g-eb7li037lghr9j81">The argument parser supports now advanced options, which are solely displayed on an extended help page of an application. </span></div>
<div id="magicdomid163" class=""><span class="author-g-eb7li037lghr9j81">A special minimiser-hash view was added to transform a sequence into a minimiser sequence of the hashes.</span></div>
<div id="magicdomid164" class=""><span class="author-g-eb7li037lghr9j81">For more details and changes please review our <a href="https://docs.seqan.de/seqan/3.0.2/about_changelog.html">changelog document</a></span><span class="author-g-eb7li037lghr9j81">.</span></div>
<div id="magicdomid165" class=""></div>
<div></div>
<div id="magicdomid166" class=""><span class="author-g-eb7li037lghr9j81">Thank you very much for subscribing and enjoy the new version SeqAn 3.0.2! </span></div>
<div id="magicdomid167" class=""></div>
<div id="magicdomid168" class=""><span class="author-g-eb7li037lghr9j81">Your SeqAn Team</span></div>
<div id="magicdomid169" class=""></div>
<div id="magicdomid170" class=""></div>
<div id="magicdomid171" class=""></div>
<p>
</div><div class="fusion-clearfix"></div></div></div></div></div>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>Raptor released</title>
		<link>https://www.seqan.de/raptor-released/</link>
		
		<dc:creator><![CDATA[Knut Reinert]]></dc:creator>
		<pubDate>Fri, 09 Oct 2020 07:02:37 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=13295</guid>

					<description><![CDATA[The SeqAn team just finished working on their new tool Raptor, a new approach for distributing approximate searches. Find the preprint at BioRXiv It is another tool in the DREAM framework moving it closer to completion (see also DREAM-Yara). In comparison with similar tools like Mantis and COBS, Raptor is 12 - 144 times faster [...]]]></description>
										<content:encoded><![CDATA[<p>The SeqAn team just finished working on their new tool Raptor, a new approach for distributing approximate searches.<br />
Find the preprint at <a href="https://www.biorxiv.org/content/10.1101/2020.10.08.330985v1">BioRXiv</a> It is another tool in the DREAM framework moving it closer to completion (see also <a href="https://academic.oup.com/bioinformatics/article/34/17/i766/5093228">DREAM-Yara</a>).</p>
<p>In comparison with similar tools like Mantis and COBS, Raptor is 12 &#8211; 144 times faster and uses up to 30 times less memory. Raptor uses winnowing minimizers to define a set of representative k-mers, an extension of the Interleaved Bloom Filters (IBF) as a set membership data structure, and probabilistic thresholding for minimizers. Our approach allows compression and a partitioning of the IBF to enable the effective use of secondary memory.</p>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn 3.0.1 released</title>
		<link>https://www.seqan.de/seqan-3-0-1-released/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Thu, 23 Jan 2020 10:54:10 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12952</guid>

					<description><![CDATA[We are excited to present a new update of our SeqAn library. This release has been in the making for roughly half a year now and we are proud to present some great new features and also a lot of improvements with respect to run time and usability. You can find a comprehensive list of [...]]]></description>
										<content:encoded><![CDATA[
<p>We are excited to present a new update of our SeqAn library. This release has been in the making for roughly half a year now and we are proud to present some great new features and also a lot of improvements  with respect to run time and usability. You can find a comprehensive list of the changes in our <a href="http://docs.seqan.de/seqan/3.0.1/about_changelog.html#autotoc_md198">changelog</a>.</p>



<ul><li>Get to know SeqAn3 with our <a href="http://docs.seqan.de/seqan/3-master-user/usergroup1.html">tutorials</a>.</li><li>Visit our <a href="http://docs.seqan.de/seqan/3.0.1/index.html">API documentation</a>.</li><li>See the <a href="http://docs.seqan.de/seqan/3-master-user/howto_porting.html">porting guide</a> for some help on porting from SeqAn2.</li></ul>



<p>Note that 3.1.0 will be the first API stable release and interfaces in this release might still change.</p>



<h3><img src="https://s.w.org/images/core/emoji/13.0.1/72x72/1f389.png" alt="🎉" class="wp-smiley" style="height: 1em; max-height: 1em;" /> Notable new features</h3>



<ul><li>
We added support for type erasing <a href="http://docs.seqan.de/seqan/3.0.1/interfaceseqan3_1_1semialphabet.html">semialphabets</a>
 that allows you to manage semialphabets with the same alphabet size in 
one container. This can have a big effect on your compile-time, in case 
you don&#8217;t drink as much <img src="https://s.w.org/images/core/emoji/13.0.1/72x72/2615.png" alt="☕" class="wp-smiley" style="height: 1em; max-height: 1em;" /> as we do.
</li><li>
We added parallel support for the alignment algorithm. You can now 
configure the number of threads you want to use for the alignment 
computation.
Click for an example 


</li><li>
One to command them all: Our argument parser now supports <strong>subcommands</strong>, such as <code>git pull</code>. A <a href="https://docs.seqan.de/seqan/3.0.1/subcommand_arg_parse.html">How-to</a> will guide you through setting this up for your tool.
</li><li>
The performance of the <strong>I/O was improved</strong> to allow faster file reading. Further, we added support for reading and writing the CIGAR string through alignment files.
</li><li>
We added several new ranges and views. Most notably, the <code>seqan3::views::kmer_hash</code> view, which transforms a sequence into a range of k-mer hashes efficiently. Another view of great practice is the <code>seqan3::views::to</code>, which can be used to convert a view into a container. We also added a <code>seqan3::dynamic_bitset</code> which is a dynamic version of the <code>std::bitset</code>.
</li><li>
Memory consumption of the (bidirectional) FM-Index for text collections was reduced by 10%.
</li></ul>



<h3>Notable API changes</h3>



<p>As much as we&#8217;d like to reduce inconsistencies between releases, we 
are sometimes forced to change an interface either to improve usability 
or to follow changes made by the ISO C++ committee.</p>



<ul><li>All our concepts are named in the <code>snake_case</code> style (e.g. <code>seqan3::WritableAlphabet</code> -&gt; <code>seqan3::writable_alphabet</code>)!</li><li>The directory <code>seqan3/range/view</code> has been renamed to <code>seqan3/range/view<strong>s</strong></code>.</li><li>The namespace <code>seqan3::view</code> has been renamed to <code>seqan3::view<strong>s</strong></code>.</li><li>The CMake variable <code>SEQAN3_VERSION_STRING</code> defined by <code>find_package(SEQAN3)</code> was renamed to <code>SEQAN3_VERSION</code>.</li></ul>



<p>You can find a comprehensive list of the changes in our <a href="http://docs.seqan.de/seqan/3.0.1/about_changelog.html#autotoc_md198">changelog</a>.</p>



<h3><img src="https://s.w.org/images/core/emoji/13.0.1/72x72/1f41b.png" alt="🐛" class="wp-smiley" style="height: 1em; max-height: 1em;" /> Notable bug fixes</h3>



<ul><li>Copying and moving the <code>seqan3::fm_index</code> and <code>seqan3::bi_fm_index</code> now work properly.</li><li>The translation table for nucleotide to amino acid translation was corrected.</li><li>The amino acid score matrices were corrected.</li></ul>



<h3><img src="https://s.w.org/images/core/emoji/13.0.1/72x72/1f50c.png" alt="🔌" class="wp-smiley" style="height: 1em; max-height: 1em;" /> External dependencies</h3>



<ul><li>We now support ranges-v3 versions &gt;= 0.10.0 and &lt; 0.11.0, 
increasing the previous requirement of &gt;= 0.5.0 and &lt; 0.6.0.</li><li>We now support cereal version 1.3.0, increasing the previous requirement of 1.2.2</li></ul>



<h3 id="Feedback"><img src="https://s.w.org/images/core/emoji/13.0.1/72x72/270d.png" alt="✍" class="wp-smiley" style="height: 1em; max-height: 1em;" /> Feedback</h3>



<p>We would love to hear what you think about the features. Report a problem or provide suggestions if you have ideas on things you’d like to see us working on. You can contact us directly <a href="https://twitter.com/SeqAnLib">@SeqAnLib</a>,&nbsp;<a href="https://gitter.im/seqan/Lobby?utm_source=badge&amp;utm_medium=badge&amp;utm_campaign=pr-badge">gitter</a>,&nbsp;<a href="https://github.com/seqan/seqan3/issues">github</a>&nbsp;or write us an email at seqan-dev <div class="fusion-fullwidth fullwidth-box fusion-builder-row-2 hundred-percent-fullwidth non-hundred-percent-height-scrolling"  style='background-color: rgba(255,255,255,0);background-position: center center;background-repeat: no-repeat;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px;'><div class="fusion-builder-row fusion-row "><div  class="fusion-layout-column fusion_builder_column fusion_builder_column_1_1 fusion-builder-column-1 fusion-one-full fusion-column-first fusion-column-last fusion-column-no-min-height 1_1"  style='margin-top:0px;margin-bottom:0px;'><div class="fusion-column-wrapper" style="background-position:left top;background-repeat:no-repeat;-webkit-background-size:cover;-moz-background-size:cover;-o-background-size:cover;background-size:cover;"   data-bg-url="">[at] lists.fu-berlin.de.</p>
<div class="fusion-clearfix"></div></div></div></div></div>]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>Announcing SeqAn3</title>
		<link>https://www.seqan.de/announcing-seqan3/</link>
		
		<dc:creator><![CDATA[Hannes Hauswedell]]></dc:creator>
		<pubDate>Thu, 06 Jun 2019 18:25:43 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12735</guid>

					<description><![CDATA[We are excited to announce version 3 of the SeqAn library. It has been many years in the making and is the largest change in the development history of SeqAn. SeqAn3 is in fact an entirely new library! Get it while it's hot: https://github.com/seqan/seqan3 Read our new documentation: http://docs.seqan.de/seqan/3-master-user/ What's new? The design goals of [...]]]></description>
										<content:encoded><![CDATA[<p><strong>We are excited to announce version 3 of the SeqAn library. It has been many years in the making and is the largest change in the development history of SeqAn. SeqAn3 is in fact an entirely new library!</strong></p>
<p><strong>Get it while it&#8217;s hot:</strong> <a href="https://github.com/seqan/seqan3" rel="noopener noreferrer" target="_blank">https://github.com/seqan/seqan3</a><br />
<strong>Read our new documentation:</strong> <a href="http://docs.seqan.de/seqan/3-master-user/" rel="noopener noreferrer" target="_blank">http://docs.seqan.de/seqan/3-master-user/</a></p>
<h2>What&#8217;s new?</h2>
<p>The design goals of SeqAn3 remain the same as for SeqAn1 and SeqAn2: be a <strong>fast, efficient, extensible C++ header library</strong> for sequence analysis. But the core design patterns for achieving these goals have changed significantly. SeqAn3 embraces &#8220;Modern C++&#8221; fully, i.e. the entire library is designed to make use of recent developments in the C++ language and standard library. In some cases we have decided to rely on third party libraries so that we don&#8217;t have to reinvent the wheel where existing solutions are already well-proven. This allows us to replace many of the more technical (and partly obscure) implementation details with existing code and focus our work on the actual algorithms and data structures that we need to solve biological problems.</p>
<p>As part of this strategy shift we are announcing a <strong>cooperation with the Succinct Data Structure Library</strong> (<a href="https://github.com/simongog/sdsl-lite">SDSL</a>), of which version 3 is a hard requirement for using SeqAn3. The SDSL provides highly optimised data structures, including (compressed) bitvectors and FM indexes. To ease integration and redistribution together with SeqAn3, the SDSL has changed their license from the LGPL to MIT and is now also a header-only library. SeqAn project members are now main contributors to the SDSL. More on the changes in the SDSL will be announced through their respective channels soon.</p>
<p>Regarding <strong>Modern C++</strong>: SeqAn3 is built on the C++17 standard and parts of the upcoming C++20 standard, including C++ Concepts and C++ Ranges. SeqAn3 is modeled strongly after the standard library, adopting many of its design patterns and code style. It is inspired by many state-of-the-art libraries like <a href="https://github.com/ericniebler/range-v3/" rel="noopener noreferrer" target="_blank">range-v3</a> and <a href="https://abseil.io/" rel="noopener noreferrer" target="_blank">abseil</a>; and it follows many C++ paradigms and best-practices developed in recent years. This results in a much <strong>cleaner, smaller and safer code base</strong>, that is easier to develop, read and maintain. We have put extra effort in providing extensive documentation, in fact there are more lines of documentation than lines of code.</p>
<p><strong>C++ Concepts</strong> allow us to move away from &#8220;template subclassing&#8221; (the mechanism used for polymorphism in SeqAn2) and enables us to use data types from outside our library as first class citizens. As a developer using SeqAn this means you finally get readable error messages when misusing an interface and you can safely <strong>use the STL&#8217;s containers or your own datatypes</strong> without any wrappers or glue involved (as long as these types model the respective concepts).</p>
<p><strong>C++ Ranges</strong> on the other hand generalise the notion of iterable objects (like containers) to types that perform certain algorithms when being iterated over. Since we deal with biological <em>sequences</em> in almost all contexts, this has a huge impact on data handling. In SeqAn3 you can create <em>views</em> on sequences that revert, complement, truncate or trim the underlying sequence without copying any data. You can even create the six-frame protein translation of a nucleotide sequence as a view that appears just like if you had generated all frames while in fact symbols are always generated on-demand and never use up memory.<br />
Files are also ranges in SeqAn3 and even algorithms, like the pairwise sequence alignment, return a dynamic range of the results that are generated while you iterate over them. </p>
<h2>Some Examples</h2>
<p><strong>Complement and translate a sequence without copying any data:</strong></p>
<pre>
std::vector<dna5> vec{"ACGTACGTACGTA"_dna5};
// pipe the vector through two view adaptors:
auto v = vec | view::complement | view::translate_single; 
// v is a view of length 4, accessing the elements on-demand will return <div class="fusion-fullwidth fullwidth-box fusion-builder-row-3 hundred-percent-fullwidth non-hundred-percent-height-scrolling"  style='background-color: rgba(255,255,255,0);background-position: center center;background-repeat: no-repeat;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px;'><div class="fusion-builder-row fusion-row "><div  class="fusion-layout-column fusion_builder_column fusion_builder_column_1_1 fusion-builder-column-2 fusion-one-full fusion-column-first fusion-column-last fusion-column-no-min-height 1_1"  style='margin-top:0px;margin-bottom:0px;'><div class="fusion-column-wrapper" style="background-position:left top;background-repeat:no-repeat;-webkit-background-size:cover;-moz-background-size:cover;-o-background-size:cover;background-size:cover;"   data-bg-url="">[C,M,H,A]
</pre>
<p><strong>Read a FastA file and print the entries:</strong></p>
<pre>
sequence_file_input fin{"my.fasta"};
// iterate over every record in the file and split it into fields:
for (auto & [ seq, id, qual ] : fin)
{
    // print the fields:
    debug_stream << "ID:  " << id << '\n';
    debug_stream << "SEQ: " << seq << '\n';
    debug_stream << "QUAL:" << qual << '\n'; // qual is empty for FastA files
}
</pre>
<p><strong>File format conversion in one line:</strong></p>
<pre>
sequence_file_input{"input.fastq"} | sequence_file_output{"output.fasta"};
// no variables are created here, all input is immediately streamed to the output
// the format is converted on-the-fly
</pre>
<h2>(Before you) Get started</h2>
<p>We believe SeqAn3 will lower the bar for writing bioinformatics applications in C++ significantly, and that the overall experience will be much improved for all developers and users of the library. However we want to point out early on that:</p>
<ul>
<li>It will require substantial work to port applications from SeqAn2 to SeqAn3, because SeqAn3 is an entirely new library.</li>
<li>The initial release does not contain all the features of SeqAn2; many will be added in the next months, but likely not all.</li>
<li>Due to relying on very Modern C++, we currently only support the GCC compiler in version 7 or newer. Experimental Clang support is being worked on; but it will likely be another year or two before Visual Studio can build apps made with SeqAn3.</li>
<li>The first API stable release will be SeqAn-3.1.0, i.e. interfaces in the 3.0 release might change slightly (although we try to avoid it).</li>
</ul>
<p><strong>Follow the development (and star us!) on GitHub:</strong> <a href="https://github.com/seqan/seqan3" rel="noopener noreferrer" target="_blank">https://github.com/seqan/seqan3</a><br />
<strong>Do the beginner tutorial:</strong> <a href="http://docs.seqan.de/seqan/3-master-user/usergroup1.html" rel="noopener noreferrer" target="_blank">http://docs.seqan.de/seqan/3-master-user/usergroup1.html</a><div class="fusion-clearfix"></div></div></div></div></div></p>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn UM 2018</title>
		<link>https://www.seqan.de/seqan-um-2018/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Tue, 07 Aug 2018 14:36:31 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12791</guid>

					<description><![CDATA[[18.08.2018: Registration is now open!] Dear SeqAn followers and bioinformatics enthusiasts, We are delighted to announce this year’s SeqAn User Meeting 2018 from 19th to 21st of September in Berlin. For the second time we will host it as a common User Meeting of the de.NBI Center for Integrative Bioinformatics (CIBI) and cordially invite researchers in the [...]]]></description>
										<content:encoded><![CDATA[<h2><span style="color: #ff0000;"><div class="fusion-fullwidth fullwidth-box fusion-builder-row-4 hundred-percent-fullwidth non-hundred-percent-height-scrolling"  style='background-color: rgba(255,255,255,0);background-position: center center;background-repeat: no-repeat;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px;'><div class="fusion-builder-row fusion-row "><div  class="fusion-layout-column fusion_builder_column fusion_builder_column_1_1 fusion-builder-column-3 fusion-one-full fusion-column-first fusion-column-last fusion-column-no-min-height 1_1"  style='margin-top:0px;margin-bottom:0px;'><div class="fusion-column-wrapper" style="background-position:left top;background-repeat:no-repeat;-webkit-background-size:cover;-moz-background-size:cover;-o-background-size:cover;background-size:cover;"   data-bg-url="">[18.08.2018: <a style="color: #ff0000;" href="https://sites.google.com/view/cibi-user-meeting-2018/home/registration">Registration</a> is now open!]</span></h2>
<p>Dear SeqAn followers and bioinformatics enthusiasts,</p>
<p>We are delighted to announce this year’s SeqAn User Meeting 2018 from 19<sup>th </sup>to 21<sup>st </sup>of September in Berlin. For the second time we will host it as a common User Meeting of the de.NBI Center for Integrative Bioinformatics (CIBI) and cordially invite researchers in the fields of genomics, proteomics, metabolomics, data mining and bioimage analysis to come together and share their knowledge.</p>
<p>We are excited to demonstrate for the first time the brand-new design of SeqAn3. In this context, we have a dedicated training day for beginners and advanced users on implementing applications with SeqAn3. With the help of your feedback and our experiences from past years we are convinced that SeqAn3 will make it genuinely easier to write applications with our library, while still providing the familiar performance of our algorithms and data structures.</p>
<p>Furthermore, we will have a training course for data scientists to learn how to process genomics data within KNIME using SeqAn and external NGS tools and the KNIME REST Web service functionality to query information from online web services such as ENSEMBL.</p>
<p>Find out more about the conference  <a href="https://sites.google.com/view/cibi-user-meeting-2018/home">here</a> or register directly <a href="https://sites.google.com/view/cibi-user-meeting-2018/home/registration">here</a>.</p>
<p>Hope to see you soon!</p>
<p>Your SeqAn Team<div class="fusion-clearfix"></div></div></div></div></div></p>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn 3 @GCB 2018</title>
		<link>https://www.seqan.de/seqan_3_tutorial_at_the_gcb_2018/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Tue, 31 Jul 2018 12:14:14 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12783</guid>

					<description><![CDATA[We are happy to announce our tutorial, SeqAn3 – a modern C++ library for efficient sequence analysis, in a full day tutorial at the GCB 2018 in Vienna. This is a beginner tutorial and focuses on tool programming using modern C++ and the SeqAn 3 library. SeqAn 3 is a major redesign of our sequence analysis [...]]]></description>
										<content:encoded><![CDATA[<p>We are happy to announce our tutorial, <strong>SeqAn3 – a modern C++ library for efficient sequence analysis</strong>, in a full day tutorial at the GCB 2018 in Vienna.</p>
<p>This is a beginner tutorial and focuses on tool programming using modern C++ and the SeqAn 3 library. SeqAn 3 is a major redesign of our sequence analysis library based on modern C++ features such as Concepts and Ranges. You will learn all about these new cool features and how tool programming is simplified using SeqAn 3.</p>
<p>You can find more information regarding the tutorial <a href="https://gcb2018.csb.univie.ac.at/seqan3">at the GCB 2018</a>.</p>
<p>Please register for the tutorials directly at the GCB 2018: <a href="https://gcb2018.csb.univie.ac.at/registration">https://gcb2018.csb.univie.ac.at/registration</a></p>
<p>Please don&#8217;t hesitate writing us if you have more questions.</p>
<p>We are looking forward meeting you there.</p>
<p>p.s. If you can&#8217;t make it to the GCB 2018, we will present the brand new SeqAn 3 design at our own User Meeting from September 19th &#8211; 21st in Berlin, together with a lot of show cases, programming and KNIME tutorials, and many great news for our users, regarding performance and efficiency. More information to our User Meeting will follow soon.</p>
<p>Also notify your colleagues about the opportunity to meet us at the <a href="http://www.seqan.de/seqan-programming-and-user-tutorial-at-the-eccb-2018/">EECB 2018</a>.</p>
<p>&nbsp;</p>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn @ECCB 2018</title>
		<link>https://www.seqan.de/seqan-programming-and-user-tutorial-at-the-eccb-2018/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Tue, 31 Jul 2018 12:05:06 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12779</guid>

					<description><![CDATA[We are happy to announce our tutorial, From an application to a fully integrated workflow – Comprehensive software engineering with SeqAn, in a full day tutorial at the ECCB 2018 in Athen. This is a beginner tutorial and focuses on basic tool programming using C++ and the SeqAn library. In addition participants will learn how SeqAn [...]]]></description>
										<content:encoded><![CDATA[<p>We are happy to announce our tutorial, <strong>From an application to a fully integrated workflow – Comprehensive software engineering with SeqAn</strong>, in a full day tutorial at the ECCB 2018 in Athen.</p>
<p>This is a beginner tutorial and focuses on basic tool programming using C++ and the SeqAn library. In addition participants will learn how SeqAn and external  tools can be integrated into the Workflowsystem KNIME to build up comprehensive workflows connected to the strong capabilities of data mining included in KNIME.</p>
<p>You can find more information regarding the tutorial <a href="http://eccb18.org/tutorial-9/">at the ECCB</a>.</p>
<p>Please register for tutorial <strong>t9</strong> directly at the ECCB 2018: <a href="http://eccb18.org/registration/">http://eccb18.org/registration/</a></p>
<p>Please don&#8217;t hesitate writing us if you have more questions.</p>
<p>We are looking forward meeting you there.</p>
<p>p.s. If you can&#8217;t make it to ECCB 2018, we will present the brand new SeqAn 3 design at the <a href="https://gcb2018.csb.univie.ac.at/seqan3">GCB 2018</a>, and we will have our own User Meeting from September 19th &#8211; 21st, with a lot of show cases, programming and KNIME tutorials, and many great news for our users, regarding performance and efficiency. More information to our User Meeting will follow soon.</p>
<p>&nbsp;</p>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn 2.4 released</title>
		<link>https://www.seqan.de/seqan-2-4-released/</link>
		
		<dc:creator><![CDATA[Hannes Hauswedell]]></dc:creator>
		<pubDate>Fri, 02 Feb 2018 20:00:17 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12732</guid>

					<description><![CDATA[We are pleased to announce the release of version 2.4 of the SeqAn library and updated versions of the official SeqAn-based applications. The alignment module has received significant updates and a parallised interface. Beyond simple multi-threading the alignment module can utilise the vector extensions of modern CPUs (SSE4, AVX2, AVX512) resulting in very impressive speed-ups, [...]]]></description>
										<content:encoded><![CDATA[<p>We are pleased to announce the release of version 2.4 of the SeqAn library and updated versions of the official SeqAn-based applications.</p>
<p>The alignment module has received significant updates and a parallised interface. Beyond simple multi-threading the alignment module can utilise the vector extensions of modern CPUs (SSE4, AVX2, AVX512) resulting in very impressive speed-ups, especially on long sequences. In fact speed-ups of ~2000x over the serial case have been measured (AVX512 and 40 threads VS no vectorisation and one thread). The details are due to be published soon.</p>
<p>The indexing module now also offers a parallelised interface which distributes multiple searches over the available CPU cores. Furthermore &#8220;optimal search schemes&#8221; can now be used for performing even faster searches with multiple errors. A preprint documenting these algorithms is available here:<br />
<a href="https://arxiv.org/abs/1711.02035" rel="noopener" target="_blank">https://arxiv.org/abs/1711.02035</a></p>
<p>The ReducedAminoAcid module now offers more reductions based on more recent research and VCF I/O is now compliant to version 4.2 of the specification, i.e. rows with only eight fields are accepted.</p>
<p>Major work went into fixing SeqAn platform support on non-canonical and big endian platforms, we are proud that also binary formats like BAM pass all tests on platforms like PowerPC64 and ARM64. All main release architectures of Debian GNU/Linux are now well supported and automatically tested. Sparc64 is one of the few platforms known not to work.</p>
<p>We have improved compatibility with new compilers and the C++17 standard so SeqAn should generate no warnings in 2018, even with stricter warning levels.</p>
<p>Finally we would like to announce that is going to be the last feature release of SeqAn2. Most of our effort is going into the development of SeqAn3, now, another announcement will cover this shortly. Patch-level releases of SeqAn2 may, however, be released if we find serious bugs or regressions.</p>
<p>Further links:</p>
<ul>
<li><a href="https://github.com/seqan/seqan/blob/master/CHANGELOG.rst" target="_blank">Full Changelog</a></li>
<li><a href="http://packages.seqan.de/" target="_blank">Downloads</a> (also applications)</li>
<li><a href="https://www.seqan.de" target="_blank">Homepage</a></li>
</ul>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn User Meeting 2017</title>
		<link>https://www.seqan.de/seqan-ugm-2017/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Fri, 14 Jul 2017 08:11:51 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12686</guid>

					<description><![CDATA[[01.08.2017] Registration now open until 15th of August!!! Joint CIBI user meeting featuring: SeqAn:  C++ library of efficient algorithms and data structures for the analysis of biological sequence data. and OpenMS + MetFrag: High-­performance software for high-­throughput proteomics and metabolomics Dear friends and colleagues, we are happy to announce the 10th CIBI User Meeting on 25. September (09:00) until [...]]]></description>
										<content:encoded><![CDATA[<style>
table, th, td {
   border: 2px solid black;
   border-collapse: collapse;
}
th {
    text-align: center;
}
</style>
<h1 style="text-align: center;" data-fontsize="34" data-lineheight="48"><span style="color: #ff0000;"><strong><div class="fusion-fullwidth fullwidth-box fusion-builder-row-5 hundred-percent-fullwidth non-hundred-percent-height-scrolling"  style='background-color: rgba(255,255,255,0);background-position: center center;background-repeat: no-repeat;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px;'><div class="fusion-builder-row fusion-row "><div  class="fusion-layout-column fusion_builder_column fusion_builder_column_1_1 fusion-builder-column-4 fusion-one-full fusion-column-first fusion-column-last fusion-column-no-min-height 1_1"  style='margin-top:0px;margin-bottom:0px;'><div class="fusion-column-wrapper" style="background-position:left top;background-repeat:no-repeat;-webkit-background-size:cover;-moz-background-size:cover;-o-background-size:cover;background-size:cover;"   data-bg-url="">[01.08.2017] Registration now open until 15th of August!!!</strong></span></h1>
<h1 style="text-align: center;" data-fontsize="34" data-lineheight="48"><strong>Joint CIBI user meeting featuring:</strong></h1>
<h2><strong>SeqAn:<em>  C++ library of efficient algorithms and data structures for the analysis of biological sequence data.</em></strong></h2>
<h2>and</h2>
<h2><strong>OpenMS + MetFrag: <em>High-­performance software for high-­throughput proteomics and metabolomics</em></strong></h2>
<h3 data-fontsize="20" data-lineheight="20">Dear friends and colleagues,</h3>
<p>we are happy to announce the 10<sup>th</sup> CIBI User Meeting on 25. September (09:00) until 27. September 2017 (16:00) at Martin-Luther University, Steintor-Campus, Room SR7, Emil-Abderhalden-Straße, 06108 Halle, Germany</p>
<p><strong>Note: SeqAn, OpenMS, and MetFrag are developed by the Center for Integrative Bioinformatics (CIBI) and funded as a project of the german</strong><strong> network for bioinformatic infrastructure (de.NBI). Thus, for the first time, the SeqAn user </strong><strong>meeting is part of a larger CIBI user meeting. Participants may select if they want to participate in the SeqAn or OpenMS track. We also offer a dedicated track for users that want to develop computational tools with SeqAn and OpenMS as well as an introduction to MetFrag.</strong></p>
<p>During the meeting, we will present the capabilities of CIBI tools SeqAn, OpenMS, and MetFrag and discuss your needs and future directions in sequence analysis and mass spectrometry.<br />
This year’s meeting will include an<strong> optional SeqAn tutorial session on the first day (Monday, September 25)</strong>  with the <strong>general conference starting on Tuesday, September 26</strong>.</p>
<p>Traditionally, the schedule of our meetings is very flexible, allowing discussions and contributions from the participants.</p>
<p>You can <strong>bring your data</strong>, a <strong>wish list</strong>, <strong>ideas for collaboration projects</strong>, and also <strong>give a talk about your project</strong>. This may include suggestions in areas where currently available software solutions were unsatisfactory. We would be very happy to welcome you in Halle this year.</p>
<h3 data-fontsize="20" data-lineheight="20">Accommodation</h3>
<p>Please take care of room reservations yourself.</p>
<p>We can suggest the following options. Prices etc. are only indicative and not guaranteed.</p>
<ul>
<li>Hotel Am Steintor (close to Workshop location)
<ul>
<li><a href="https://www.am-steintor.de/">https://www.am-steintor.de</a></li>
<li>40 EUR single room, 55 EUR double room</li>
</ul>
</li>
<li>Hostel No5 (20 min walk, or Tram connection, Nr 2 and Nr 5)
<ul>
<li><a href="http://www.hostel-no5.de/hostel_lage1.html">http://www.hostel-no5.de/hostel_lage1.html</a></li>
<li>single room: 25 EUR per night | double room: 40 EUR per night | triple room: 15 EUR per night and person</li>
<li>studio (single room, separate bath, kitchen): 40 EUR per night (single) or 50 EUR per night (double)</li>
<li>plus breakfast: 4 EUR per day and person (must be ordered before arrival)</li>
</ul>
</li>
<li>Hotel Ankerhof (20 min walk or direct Tram connection, Nr 2 and Nr 5)
<ul>
<li><a href="http://www.leipziggrafik.de/anker05/htdocs/anfahrt.shtml">http://www.leipziggrafik.de/anker05/htdocs/anfahrt.shtml</a></li>
<li>special rate (incl. breakfast) is 65 EUR per person and night with booking code: «Leibniz Institut für Pflanzenbiochemie»</li>
</ul>
</li>
<li>Further accommodation options:
<ul>
<li><a href="http://www.hrs.de/">http://www.hrs.de</a></li>
<li><a href="https://www.couchsurfing.org/">https://www.couchsurfing.org/</a></li>
</ul>
</li>
</ul>
<h3 data-fontsize="20" data-lineheight="20">Journey</h3>
<p><b>Directions to the Steintor Campus Uni Halle: </b><a href="https://goo.gl/ZNSmBM"><b>https://goo.gl/ZNSmBM</b></a></p>
<p><b>By train:</b> Halle is well connected within Germany’s network of IC and ICE trains <a href="https://reiseauskunft.bahn.de/bin/query2.exe/en">https://reiseauskunft.bahn.de/bin/query2.exe/en</a></p>
<p><b>By plane: </b>Nearest airports are Halle-Leipzig (LEJ), Erfurt (ERF), Berlin (TXL &amp; SXF).</p>
<p>Shortest connection is via Halle-Leipzig Airport, but flights might not run every day. Nice connections are possible via Berlin with train to Halle (Saale) and via Frankfurt (FRA) using the ICE train connection. Note that the Berlin airport BER is still not opened.</p>
<p><b>Public transport in Halle:</b> <a href="https://goo.gl/9Zo5vy">https://goo.gl/9Zo5vy</a>, <a href="http://www.havag.com/fahrplan/liniennetz-und-tarifzonenplan">http://www.havag.com/fahrplan/liniennetz-und-tarifzonenplan</a></p>
<h3 data-fontsize="20" data-lineheight="20"><strong>Registration</strong></h3>
<p>Participants need to register via <a href="https://goo.gl/forms/8WKW2T2OBrosfiuA3">https://goo.gl/forms/8WKW2T2OBrosfiuA3</a>. The deadline for the workshop registration is <strong>15</strong><b>. August 2017</b>. Notification of acceptance is <strong>31</strong><b>. August 2017</b>. The workshop fee is 50 € for students (Undergrad / PhD), and 80 € for Industry, PostDoc, PI, Professor or above. Fees will cover coffee breaks, lunch, and the workshop dinner.</p>
<h3 data-fontsize="20" data-lineheight="20">Schedule</h3>
<table>
<thead>
<tr>
<th rowspan="2"></th>
<th colspan="3">Monday 2017-09-25</th>
<th>Tuesday 2017-09-26</th>
<th>Wednesday 2017-09-27</th>
</tr>
<tr>
<th>Room: SR7</th>
<th>Room: SR12</th>
<th>Room: SR8</th>
<th>Room: SR7</th>
<th>Room: SR7</th>
</tr>
</thead>
<tbody>
<tr>
<td>09:00 &#8211; 09:30</td>
<td colspan="3" align="center">Welcome and Introduction</td>
<td>Introduction to CIBI and its Tools<br />
Kohlbacher</p>
<p>What&#8217;s new in OpenMS<br />
Kohlbacher</td>
<td>Developer Talks</td>
</tr>
<tr>
<td>09:30 &#8211; 10:30</td>
<td>Workflow Tutorial<br />
with KNIME</td>
<td>Introduction to<br />
Programming with<br />
SeqAn</td>
<td>Introduction to<br />
Programming with<br />
OpenMS</td>
<td>What&#8217;s new in &#8230;</p>
<p>&#8230; SeqAn?<br />
&#8230; KNIME?<br />
&#8230; MetFrag?</td>
<td>Scientiffic talks</td>
</tr>
<tr>
<td>10:30 &#8211; 11:00</td>
<td colspan="3" align="center"><span style="color: #009999;">Coffee Break</span></td>
<td align="center"><span style="color: #009999;">Coffee Break</span></td>
<td align="center"><span style="color: #009999;">Coffee Break</span></td>
</tr>
<tr>
<td>11:00 &#8211; 12:00</td>
<td>Introduction to OpenMS<br />
and TOPP</td>
<td>Intro to MetFrag<br />
&amp; MassBank</td>
<td>SeqAn Tools and<br />
KNIME Workflows</td>
<td>Keynote: Ivo Große<br />
What’s New ?</td>
<td>Keynote: Wolfram Weckwerth<br />
System theoretical concepts<br />
in multidimensional genome-scale<br />
molecular data analysis</td>
</tr>
<tr>
<td>12:00 &#8211; 13:00</td>
<td colspan="3" align="center"><span style="color: #009999;">Lunch at <a href="https://www.studentenwerk-halle.de/hochschulgastronomie/halle/harzmensa/">Harzmensa</a> (700m walk)</span></td>
<td><span style="color: #009999;">Lunch at <a href="https://www.studentenwerk-halle.de/hochschulgastronomie/halle/harzmensa/">Harzmensa</a> (700m walk)</span></td>
<td><span style="color: #009999;">Lunch at <a href="https://www.studentenwerk-halle.de/hochschulgastronomie/halle/harzmensa/">Harzmensa</a> (700m walk)</span></td>
</tr>
<tr>
<td>13:00 &#8211; 14:30</td>
<td colspan="2">Advanced OpenMS</td>
<td>Advanced SeqAn Tools</td>
<td>Developer talks:<br />
SeqAn IPCC (Rahn)<br />
dynamic KNIME nodes (Pfeuffer)<br />
OpenMS talk</td>
<td>Scientiffic talks</td>
</tr>
<tr>
<td>14:30 &#8211; 15:00</td>
<td colspan="3" align="center"><span style="color: #009999;">Coffee Break</span></td>
<td align="center"><span style="color: #009999;">Coffee Break</span></td>
<td align="center"><span style="color: #009999;">Coffee Break</span></td>
</tr>
<tr>
<td>15:00 &#8211; 16:00</td>
<td>Advanced Proteomics and<br />
Metabolomics Workflows</td>
<td>Advanced NGS<br />
Data Analysis</td>
<td>BYOD/Hacking</td>
<td>Keynote: Greg Landrum<br />
t.b.a.</td>
<td>Scientiffic talks<br />
Wrap-up and departure</td>
</tr>
<tr>
<td>16:15 &#8211; 17:30</td>
<td>Integrated Multi-Omics<br />
Workflows</td>
<td>BYOD/Hacking</td>
<td>BYOD/Hacking</td>
<td>Scientiffic talks</td>
<td></td>
</tr>
<tr>
<td>17:30 &#8211; end</td>
<td colspan="3" align="center"><span style="color: #009999;">Self-paid DINNER at <a href="http://krugzumgruenenkranze.de/">Krug zum grünen Kranze</a></span></td>
<td><span style="color: #009999;">Social event Tram City Tour<br />
and Workshop DINNER at<br />
<a href="http://www.halleschesbrauhaus.de/">Brauhaus Halle</a></span></td>
<td></td>
</tr>
</tbody>
</table>
<p>Speaker:<br />
We are excited to have Keynote talks by Wolfram Weckwerth (MOSYS, Uni Vienna), Greg Landrum (RDKIT/KNIME), Ivo Grosse (MLU Halle), and Steffen Neumann (IPB Halle)</p>
<h3 data-fontsize="20" data-lineheight="20">Contact</h3>
<p><a href="mailto:seqan-dev@lists.fu-berlin.de">SeqAn Team</a> – scientific and program questions</p>
<p>See you in Halle!<br />
The CIBi Team<div class="fusion-clearfix"></div></div></div></div></div></p>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SEQAN 2.3 RELEASED</title>
		<link>https://www.seqan.de/seqan-2-3-released/</link>
		
		<dc:creator><![CDATA[Svenja Mehringer]]></dc:creator>
		<pubDate>Fri, 16 Dec 2016 13:29:35 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12671</guid>

					<description><![CDATA[We are pleased to announce the release of version 2.3 of the SeqAn library and updated versions of the official SeqAn-based applications. Major advancements have been accomplished in the area of Input and Output, e.g. we now support I/O of RNA structure files like Vienna or Stockholm and we included handling of new fields in Blast [...]]]></description>
										<content:encoded><![CDATA[<p>We are pleased to announce the release of version 2.3 of the SeqAn library and updated versions of the official SeqAn-based applications.</p>
<p>Major advancements have been accomplished in the area of Input and Output, e.g. we now support I/O of RNA structure files like Vienna or Stockholm and we included handling of new fields in Blast files.  We further improved our bidirectional FM-Index by substantially reducing the memory footprint and we put much effort into a detailed documentation. Additionally, the SeqAn command line parser will now inform application users whenever a new version of their app is available. This requires a ping to our servers which can be disabled either at run-time or at compile time if this is a concern to you (see our <a href="http://seqan.readthedocs.io/en/master/">manual</a> for further details).</p>
<p>Besides new features, we addressed various issues that were kindly reported to us on Github. Of some interest might be the adaption of format changes in GFF/GTF, Bam and VCF files, the enforcement of container-types for the find module of indices, and the allowance of larger data in the graph module due to a non-recurse reimplementation of the depth-first-search.</p>
<p>Infrastructural changes include some improvements to CMake and PkgConfig file. SeqAn now switched to the &#8220;Config“ mode for detecting SeqAn, which is the proper form of detecting external libraries. This means that we now ship SeqAn with a &#8220;seqan-config.cmake&#8221; instead of a &#8220;FindSeqan.cmake“ file. In general this might not affect your cmake-based project at all. But in case you experience any issues you can read more about the implications in the official API documentation of the <a href="https://cmake.org/cmake/help/v3.0/command/find_package.html">find_package function</a>.</p>
<p>SeqAn is now also tested for various non-x86 CPU architectures including arm, mips, powerpc and sparc. And it even works on GNU/Hurd! Thanks to the Debian med team for their contributions.</p>
<p>All publicly documented interfaces are preserved and no supported platforms where dropped. It is our policy that this shall not happen in the 2.x cycle anymore at all so you can trust that your applications work with the next releases.</p>
<p>Further links:</p>
<ul>
<li><a href="https://github.com/seqan/seqan/blob/master/CHANGELOG.rst" target="_blank">Full Changelog</a></li>
<li><a href="http://packages.seqan.de/" target="_blank">Downloads</a> (also applications)</li>
<li><a href="https://www.seqan.de" target="_blank">Homepage</a></li>
</ul>
]]></content:encoded>
					
		
		
			</item>
	</channel>
</rss>
