<?xml version="1.0" encoding="UTF-8"?><rss version="2.0"
	xmlns:content="http://purl.org/rss/1.0/modules/content/"
	xmlns:wfw="http://wellformedweb.org/CommentAPI/"
	xmlns:dc="http://purl.org/dc/elements/1.1/"
	xmlns:atom="http://www.w3.org/2005/Atom"
	xmlns:sy="http://purl.org/rss/1.0/modules/syndication/"
	xmlns:slash="http://purl.org/rss/1.0/modules/slash/"
	>

<channel>
	<title>Rene Rahn &#8211; Seqan</title>
	<atom:link href="http://www.seqan.de/author/rene-rahn/feed/" rel="self" type="application/rss+xml" />
	<link>http://www.seqan.de</link>
	<description></description>
	<lastBuildDate>Tue, 13 Oct 2020 11:29:01 +0000</lastBuildDate>
	<language>en-US</language>
	<sy:updatePeriod>
	hourly	</sy:updatePeriod>
	<sy:updateFrequency>
	1	</sy:updateFrequency>
	<generator>https://wordpress.org/?v=5.6.1</generator>

<image>
	<url>http://www.seqan.de/wp-content/uploads/2020/11/cropped-seqan_logo_4-2-32x32.png</url>
	<title>Rene Rahn &#8211; Seqan</title>
	<link>http://www.seqan.de</link>
	<width>32</width>
	<height>32</height>
</image> 
	<item>
		<title>SeqAn 3.0.2 released</title>
		<link>http://www.seqan.de/seqan-3-0-2-release/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Tue, 13 Oct 2020 11:23:10 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=13313</guid>

					<description><![CDATA[Dear SeqAn users, supporters, and subscribers, today we present a new update of the SeqAn library. Hooray!!! The following highlights are present in the current release. General C++-20 support The C++-20 support for concepts and type-traits has been fully integrated into SeqAn. This means that even in C++-17 mode you can use the concept [...]]]></description>
										<content:encoded><![CDATA[<div class="fusion-fullwidth fullwidth-box fusion-builder-row-1 nonhundred-percent-fullwidth non-hundred-percent-height-scrolling"  style='background-color: rgba(255,255,255,0);background-position: center center;background-repeat: no-repeat;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px;'><div class="fusion-builder-row fusion-row "><div  class="fusion-layout-column fusion_builder_column fusion_builder_column_1_1 fusion-builder-column-0 fusion-one-full fusion-column-first fusion-column-last 1_1"  style='margin-top:0px;margin-bottom:0px;'><div class="fusion-column-wrapper" style="padding: 0px 0px 0px 0px;background-position:left top;background-repeat:no-repeat;-webkit-background-size:cover;-moz-background-size:cover;-o-background-size:cover;background-size:cover;"   data-bg-url=""><div class="fusion-text"></p>
<div id="magicdomid5" class=""><span class="author-g-eb7li037lghr9j81">Dear SeqAn users, supporters, and subscribers,</span></div>
<div id="magicdomid6" class=""></div>
<div id="magicdomid7" class=""><span class="author-g-eb7li037lghr9j81">today we present a new update of the SeqAn library. Hooray!!!</span></div>
<div id="magicdomid177" class="ace-line"><span class="author-g-eb7li037lghr9j81">The following highlights are present in the current release.</span></div>
<h2 id="magicdomid180" class="ace-line"><span class="author-g-eb7li037lghr9j81">General C++-20 support</span></h2>
<div id="magicdomid182" class="ace-line"><span class="author-g-eb7li037lghr9j81">The </span><span class="author-g-ojf51wntp1sx7fao">C</span><span class="author-g-eb7li037lghr9j81">++-20 support for concepts and type-traits has been fully integrated into SeqAn.</span></div>
<div id="magicdomid183" class="ace-line"><span class="author-g-eb7li037lghr9j81">This means that even in </span><span class="author-g-ojf51wntp1sx7fao">C</span><span class="author-g-eb7li037lghr9j81">++-17 mode you can use the concept names and type traits as defined by the </span><span class="author-g-ojf51wntp1sx7fao">C</span><span class="author-g-eb7li037lghr9j81">++20-specification</span></div>
<div id="magicdomid184" class="ace-line"><span class="author-g-eb7li037lghr9j81">and switch to </span><span class="author-g-ojf51wntp1sx7fao">C</span><span class="author-g-eb7li037lghr9j81">++-20 without breaking your code.</span></div>
<h2 id="magicdomid186" class="ace-line"><span class="author-g-eb7li037lghr9j81">Cookbook</span></h2>
<div id="magicdomid188" class="ace-line"><span class="author-g-eb7li037lghr9j81">We present the new <a href="https://docs.seqan.de/seqan/3.0.2/cookbook.html">Cookbook</a></span><span class="author-g-eb7li037lghr9j81">, where you can find usefulÂ </span></div>
<div id="magicdomid189" class="ace-line"><span class="author-g-eb7li037lghr9j81">code examples for typical tasks with SeqAn, e.g. reading a sequence file, or searching for a pattern in a string.Â </span></div>
<h2 id="magicdomid191" class="ace-line"><span class="author-g-eb7li037lghr9j81">Application template</span></h2>
<div id="magicdomid193" class="ace-line"><span class="author-g-eb7li037lghr9j81">In order to get started with new SeqAn3 applications easily, we introduce the newÂ </span></div>
<div id="magicdomid194" class="ace-line"><span class="author-g-eb7li037lghr9j81"><a href="https://github.com/seqan/app-template">Application Template</a></span><span class="author-g-eb7li037lghr9j81">, where you can check out a template application withÂ </span></div>
<div id="magicdomid195" class="ace-line"><span class="author-g-eb7li037lghr9j81">build system, tests and all necessary requirements set up.Â </span></div>
<h2 id="magicdomid11"><span class="author-g-eb7li037lghr9j81">User callback in alignment and search</span></h2>
<div id="magicdomid13" class=""><span class="author-g-eb7li037lghr9j81">The pairwise alignment can now be configured with a user-defined callback, which is called for every computed alignment</span></div>
<div id="magicdomid14" class=""><span class="author-g-eb7li037lghr9j81">result instead of returning a lazy range over the alignment results:</span></div>
<div></div>
<div><style type="text/css" scopped="scopped">.fusion-syntax-highlighter-1 > .CodeMirror .CodeMirror-gutters { background-color: ; }.fusion-syntax-highlighter-1 > .CodeMirror .CodeMirror-linenumber { color: ; }</style><div class="fusion-syntax-highlighter-container fusion-syntax-highlighter-1 fusion-syntax-highlighter-theme-dark" style="opacity:0;margin-top:0px;margin-right:0px;margin-bottom:0px;margin-left:0px;font-size:14px;border-width:1px;border-style:solid;border-color:#e0dede;"><div class="syntax-highlighter-copy-code"><span class="syntax-highlighter-copy-code-title" data-id="fusion_syntax_highlighter_1" style="font-size:14px;">Copy to Clipboard</span></div><textarea class="fusion-syntax-highlighter-textarea" id="fusion_syntax_highlighter_1" data-readOnly="nocursor" data-lineNumbers="1" data-lineWrapping="" data-theme="oceanic-next">#include <mutex>
#include <vector>

#include <seqan3/alignment/configuration/align_config_edit.hpp>
#include <seqan3/alignment/configuration/align_config_method.hpp>
#include <seqan3/alignment/configuration/align_config_on_result.hpp>
#include <seqan3/alignment/configuration/align_config_parallel.hpp>
#include <seqan3/alignment/pairwise/align_pairwise.hpp>
#include <seqan3/alphabet/nucleotide/dna4.hpp>
#include <seqan3/core/debug_stream.hpp>

int main()
{
    // Generate some sequences.
    using seqan3::operator""_dna4;
    using sequence_pair_t = std::pair<seqan3::dna4_vector, seqan3::dna4_vector>;
    std::vector<sequence_pair_t> sequences{100, {"AGTGCTACG"_dna4, "ACGTGCGACTAG"_dna4}};

    std::mutex write_to_debug_stream{}; // Need mutex to synchronise the output.

    // Use edit distance with 4 threads.
    auto const alignment_config = seqan3::align_cfg::method_global{} |
                                  seqan3::align_cfg::edit_scheme |
                                  seqan3::align_cfg::parallel{4} |
                                  seqan3::align_cfg::on_result{[&] (auto && result)
                                  {
                                      std::lock_guard sync{write_to_debug_stream}; // critical section
                                      seqan3::debug_stream << result << '\n';
                                  }};

    // Compute the alignments in parallel, and output them unordered using the callback (order is not deterministic).
    seqan3::align_pairwise(sequences, alignment_config);  // seqan3::align_pairwise is now declared void.
}</textarea></div></div>
<div id="magicdomid52" class=""></div>
<div id="magicdomid196" class="ace-line"><span class="author-g-eb7li037lghr9j81">This feature was also added to the `seqan3::search` algorithm, i.e. both algorithms can be customised using a user callback.</span></div>
<h2 id="magicdomid56"><span class="author-g-eb7li037lghr9j81">Lazy search result range</span></h2>
<div id="magicdomid58" class=""><span class="author-g-eb7li037lghr9j81">The return type of the search algorithm was adapted to use a lazy result range over the found hits during theÂ </span></div>
<div id="magicdomid59" class=""><span class="author-g-eb7li037lghr9j81">search. This allows you to use the same code to step through the hits independently of having an index over a single</span></div>
<div id="magicdomid60" class=""><span class="author-g-eb7li037lghr9j81">text or a text collection:</span></div>
<div id="magicdomid61" class=""></div>
<div><style type="text/css" scopped="scopped">.fusion-syntax-highlighter-2 > .CodeMirror .CodeMirror-gutters { background-color: ; }.fusion-syntax-highlighter-2 > .CodeMirror .CodeMirror-linenumber { color: ; }</style><div class="fusion-syntax-highlighter-container fusion-syntax-highlighter-2 fusion-syntax-highlighter-theme-dark" style="opacity:0;margin-top:0px;margin-right:0px;margin-bottom:0px;margin-left:0px;font-size:14px;border-width:1px;border-style:solid;border-color:#e0dede;"><div class="syntax-highlighter-copy-code"><span class="syntax-highlighter-copy-code-title" data-id="fusion_syntax_highlighter_2" style="font-size:14px;">Copy to Clipboard</span></div><textarea class="fusion-syntax-highlighter-textarea" id="fusion_syntax_highlighter_2" data-readOnly="nocursor" data-lineNumbers="1" data-lineWrapping="" data-theme="oceanic-next">#include <vector>

#include <seqan3/alphabet/nucleotide/dna4.hpp>
#include <seqan3/core/debug_stream.hpp>
#include <seqan3/search/configuration/max_error.hpp>
#include <seqan3/search/configuration/hit.hpp>
#include <seqan3/search/search.hpp>
#include <seqan3/search/fm_index/fm_index.hpp>

int main()
{
    using seqan3::operator""_dna4;

    std::vector<seqan3::dna4_vector> text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTA"_dna4,
                                          "ACCCGATGAGCTACCCAGTAGTCGAACTG"_dna4,
                                          "GGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
    seqan3::dna4_vector query{"GCT"_dna4};

    seqan3::configuration const search_config = seqan3::search_cfg::max_error_total{seqan3::search_cfg::error_count{1}} |
                                                seqan3::search_cfg::hit_all_best{};

    // Always provide a unified interface over the found hits independent of the index its text layout.
    seqan3::debug_stream << "Search in text collection:\n";
    seqan3::fm_index index_collection{text};
    for (auto && hit : search(query, index_collection, search_config)) // Over a text collection.
        seqan3::debug_stream << hit << '\n';

    seqan3::debug_stream << "\nSearch in single text:\n";
    seqan3::fm_index index_single{text[0]};
    for (auto && hit : search(query, index_single, search_config)) // Over a single text.
        seqan3::debug_stream << hit << '\n';
}
</textarea></div></div>
<h2 id="magicdomid97"><span class="author-g-eb7li037lghr9j81">Dynamic hit configuration</span></h2>
<div id="magicdomid99" class=""><span class="author-g-eb7li037lghr9j81">The hit configuration can now be set dynamically:</span></div>
<div id="magicdomid100" class=""></div>
<div><style type="text/css" scopped="scopped">.fusion-syntax-highlighter-3 > .CodeMirror .CodeMirror-gutters { background-color: ; }.fusion-syntax-highlighter-3 > .CodeMirror .CodeMirror-linenumber { color: ; }</style><div class="fusion-syntax-highlighter-container fusion-syntax-highlighter-3 fusion-syntax-highlighter-theme-dark" style="opacity:0;margin-top:0px;margin-right:0px;margin-bottom:0px;margin-left:0px;font-size:14px;border-width:1px;border-style:solid;border-color:#e0dede;"><div class="syntax-highlighter-copy-code"><span class="syntax-highlighter-copy-code-title" data-id="fusion_syntax_highlighter_3" style="font-size:14px;">Copy to Clipboard</span></div><textarea class="fusion-syntax-highlighter-textarea" id="fusion_syntax_highlighter_3" data-readOnly="nocursor" data-lineNumbers="1" data-lineWrapping="" data-theme="oceanic-next">#include <vector>

#include <seqan3/alphabet/nucleotide/dna4.hpp>
#include <seqan3/core/debug_stream.hpp>
#include <seqan3/search/configuration/max_error.hpp>
#include <seqan3/search/configuration/hit.hpp>
#include <seqan3/search/fm_index/fm_index.hpp>
#include <seqan3/search/search.hpp>

int main()
{
    using seqan3::operator""_dna4;

    std::vector<seqan3::dna4_vector> text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTA"_dna4,
                                          "ACCCGATGAGCTACCCAGTAGTCGAACTG"_dna4,
                                          "GGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
    seqan3::dna4_vector query{"GCT"_dna4};
    seqan3::fm_index index{text};

    // Use the dynamic hit configuration to set hit_all_best mode.
    seqan3::configuration search_config = seqan3::search_cfg::max_error_total{seqan3::search_cfg::error_count{1}} |
                                          seqan3::search_cfg::hit{seqan3::search_cfg::hit_all_best{}};

    seqan3::debug_stream << "All single best hits:\n";
    for (auto && hit : search(query, index, search_config)) // Find all best hits:
        seqan3::debug_stream << hit << '\n';

    // Change the hit configuration to the strata mode with a stratum of 1.
    using seqan3::get;
    get<seqan3::search_cfg::hit>(search_config).hit_variant = seqan3::search_cfg::hit_strata{1};

    seqan3::debug_stream << "\nAll x+1 hits:\n";
    for (auto && hit : search(query, index, search_config)) // Find all strata hits.
        seqan3::debug_stream << hit << '\n';
}</textarea></div></div>
<h2 id="magicdomid156"><span class="author-g-eb7li037lghr9j81">Other important changes and additions</span></h2>
<div id="magicdomid158" class=""><span class="author-g-eb7li037lghr9j81">We put much effort into harmonising the algorithm configuration interface for the pairwise alignment andÂ </span></div>
<div id="magicdomid159" class=""><span class="author-g-eb7li037lghr9j81">search algorithm. This involves some structural changes, which break the current code, but make the interfaces much moreÂ </span></div>
<div id="magicdomid160" class=""><span class="author-g-eb7li037lghr9j81">stable and easier to use.</span></div>
<div id="magicdomid203" class="ace-line"><span class="author-g-eb7li037lghr9j81">SeqAn</span><span class="author-g-l4p5hayyb3txbm6e"> n</span><span class="author-g-d4phsgoja0lb2x42">ow</span><span class="author-g-eb7li037lghr9j81"> offers</span> <span class="author-g-eb7li037lghr9j81">a data structure called interleaved Bloom filter, which can answer set-membership queries efficiently.</span></div>
<div id="magicdomid162" class=""><span class="author-g-eb7li037lghr9j81">The argument parser supports now advanced options, which are solely displayed on an extended help page of an application.Â </span></div>
<div id="magicdomid163" class=""><span class="author-g-eb7li037lghr9j81">A special minimiser-hash view was added to transform a sequence into a minimiser sequence of the hashes.</span></div>
<div id="magicdomid164" class=""><span class="author-g-eb7li037lghr9j81">For more details and changes please review our <a href="https://docs.seqan.de/seqan/3.0.2/about_changelog.html">changelog document</a></span><span class="author-g-eb7li037lghr9j81">.</span></div>
<div id="magicdomid165" class=""></div>
<div></div>
<div id="magicdomid166" class=""><span class="author-g-eb7li037lghr9j81">Thank you very much for subscribing and enjoy the new version SeqAn 3.0.2!Â </span></div>
<div id="magicdomid167" class=""></div>
<div id="magicdomid168" class=""><span class="author-g-eb7li037lghr9j81">Your SeqAn Team</span></div>
<div id="magicdomid169" class=""></div>
<div id="magicdomid170" class=""></div>
<div id="magicdomid171" class=""></div>
<p>
</div><div class="fusion-clearfix"></div></div></div></div></div>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn 3.0.1 released</title>
		<link>http://www.seqan.de/seqan-3-0-1-released/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Thu, 23 Jan 2020 10:54:10 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12952</guid>

					<description><![CDATA[We are excited to present a new update of our SeqAn library. This release has been in the making for roughly half a year now and we are proud to present some great new features and also a lot of improvements with respect to run time and usability. You can find a comprehensive list of [...]]]></description>
										<content:encoded><![CDATA[
<p>We are excited to present a new update of our SeqAn library. This release has been in the making for roughly half a year now and we are proud to present some great new features and also a lot of improvements  with respect to run time and usability. You can find a comprehensive list of the changes in our <a href="http://docs.seqan.de/seqan/3.0.1/about_changelog.html#autotoc_md198">changelog</a>.</p>



<ul><li>Get to know SeqAn3 with our <a href="http://docs.seqan.de/seqan/3-master-user/usergroup1.html">tutorials</a>.</li><li>Visit our <a href="http://docs.seqan.de/seqan/3.0.1/index.html">API documentation</a>.</li><li>See the <a href="http://docs.seqan.de/seqan/3-master-user/howto_porting.html">porting guide</a> for some help on porting from SeqAn2.</li></ul>



<p>Note that 3.1.0 will be the first API stable release and interfaces in this release might still change.</p>



<h3><img src="https://s.w.org/images/core/emoji/13.0.1/72x72/1f389.png" alt="ðŸŽ‰" class="wp-smiley" style="height: 1em; max-height: 1em;" /> Notable new features</h3>



<ul><li>
We added support for type erasing <a href="http://docs.seqan.de/seqan/3.0.1/interfaceseqan3_1_1semialphabet.html">semialphabets</a>
 that allows you to manage semialphabets with the same alphabet size in 
one container. This can have a big effect on your compile-time, in case 
you don&#8217;t drink as much <img src="https://s.w.org/images/core/emoji/13.0.1/72x72/2615.png" alt="â˜•" class="wp-smiley" style="height: 1em; max-height: 1em;" /> as we do.
</li><li>
We added parallel support for the alignment algorithm. You can now 
configure the number of threads you want to use for the alignment 
computation.
Click for an example 


</li><li>
One to command them all: Our argument parser now supports <strong>subcommands</strong>, such as <code>git pull</code>. A <a href="https://docs.seqan.de/seqan/3.0.1/subcommand_arg_parse.html">How-to</a> will guide you through setting this up for your tool.
</li><li>
The performance of the <strong>I/O was improved</strong> to allow faster file reading. Further, we added support for reading and writing the CIGAR string through alignment files.
</li><li>
We added several new ranges and views. Most notably, the <code>seqan3::views::kmer_hash</code> view, which transforms a sequence into a range of k-mer hashes efficiently. Another view of great practice is the <code>seqan3::views::to</code>, which can be used to convert a view into a container. We also added a <code>seqan3::dynamic_bitset</code> which is a dynamic version of the <code>std::bitset</code>.
</li><li>
Memory consumption of the (bidirectional) FM-Index for text collections was reduced by 10%.
</li></ul>



<h3>Notable API changes</h3>



<p>As much as we&#8217;d like to reduce inconsistencies between releases, we 
are sometimes forced to change an interface either to improve usability 
or to follow changes made by the ISO C++ committee.</p>



<ul><li>All our concepts are named in the <code>snake_case</code> style (e.g. <code>seqan3::WritableAlphabet</code> -&gt; <code>seqan3::writable_alphabet</code>)!</li><li>The directory <code>seqan3/range/view</code> has been renamed to <code>seqan3/range/view<strong>s</strong></code>.</li><li>The namespace <code>seqan3::view</code> has been renamed to <code>seqan3::view<strong>s</strong></code>.</li><li>The CMake variable <code>SEQAN3_VERSION_STRING</code> defined by <code>find_package(SEQAN3)</code> was renamed to <code>SEQAN3_VERSION</code>.</li></ul>



<p>You can find a comprehensive list of the changes in our <a href="http://docs.seqan.de/seqan/3.0.1/about_changelog.html#autotoc_md198">changelog</a>.</p>



<h3><img src="https://s.w.org/images/core/emoji/13.0.1/72x72/1f41b.png" alt="ðŸ›" class="wp-smiley" style="height: 1em; max-height: 1em;" /> Notable bug fixes</h3>



<ul><li>Copying and moving the <code>seqan3::fm_index</code> and <code>seqan3::bi_fm_index</code> now work properly.</li><li>The translation table for nucleotide to amino acid translation was corrected.</li><li>The amino acid score matrices were corrected.</li></ul>



<h3><img src="https://s.w.org/images/core/emoji/13.0.1/72x72/1f50c.png" alt="ðŸ”Œ" class="wp-smiley" style="height: 1em; max-height: 1em;" /> External dependencies</h3>



<ul><li>We now support ranges-v3 versions &gt;= 0.10.0 and &lt; 0.11.0, 
increasing the previous requirement of &gt;= 0.5.0 and &lt; 0.6.0.</li><li>We now support cereal version 1.3.0, increasing the previous requirement of 1.2.2</li></ul>



<h3 id="Feedback"><img src="https://s.w.org/images/core/emoji/13.0.1/72x72/270d.png" alt="âœ" class="wp-smiley" style="height: 1em; max-height: 1em;" /> Feedback</h3>



<p>We would love to hear what you think about the features. Report a problem or provide suggestions if you have ideas on things youâ€™d like to see us working on. You can contact us directly <a href="https://twitter.com/SeqAnLib">@SeqAnLib</a>,&nbsp;<a href="https://gitter.im/seqan/Lobby?utm_source=badge&amp;utm_medium=badge&amp;utm_campaign=pr-badge">gitter</a>,&nbsp;<a href="https://github.com/seqan/seqan3/issues">github</a>&nbsp;or write us an email at seqan-dev <div class="fusion-fullwidth fullwidth-box fusion-builder-row-2 hundred-percent-fullwidth non-hundred-percent-height-scrolling"  style='background-color: rgba(255,255,255,0);background-position: center center;background-repeat: no-repeat;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px;'><div class="fusion-builder-row fusion-row "><div  class="fusion-layout-column fusion_builder_column fusion_builder_column_1_1 fusion-builder-column-1 fusion-one-full fusion-column-first fusion-column-last fusion-column-no-min-height 1_1"  style='margin-top:0px;margin-bottom:0px;'><div class="fusion-column-wrapper" style="background-position:left top;background-repeat:no-repeat;-webkit-background-size:cover;-moz-background-size:cover;-o-background-size:cover;background-size:cover;"   data-bg-url="">[at] lists.fu-berlin.de.</p>
<div class="fusion-clearfix"></div></div></div></div></div>]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn UM 2018</title>
		<link>http://www.seqan.de/seqan-um-2018/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Tue, 07 Aug 2018 14:36:31 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12791</guid>

					<description><![CDATA[[18.08.2018: Registration is now open!] Dear SeqAn followers and bioinformatics enthusiasts, We are delighted to announce this yearâ€™s SeqAn User Meeting 2018 from 19thÂ to 21stÂ of September in Berlin. For the second time we will host it as a common User Meeting of the de.NBI Center for Integrative Bioinformatics (CIBI) and cordially invite researchers in the [...]]]></description>
										<content:encoded><![CDATA[<h2><span style="color: #ff0000;"><div class="fusion-fullwidth fullwidth-box fusion-builder-row-3 hundred-percent-fullwidth non-hundred-percent-height-scrolling"  style='background-color: rgba(255,255,255,0);background-position: center center;background-repeat: no-repeat;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px;'><div class="fusion-builder-row fusion-row "><div  class="fusion-layout-column fusion_builder_column fusion_builder_column_1_1 fusion-builder-column-2 fusion-one-full fusion-column-first fusion-column-last fusion-column-no-min-height 1_1"  style='margin-top:0px;margin-bottom:0px;'><div class="fusion-column-wrapper" style="background-position:left top;background-repeat:no-repeat;-webkit-background-size:cover;-moz-background-size:cover;-o-background-size:cover;background-size:cover;"   data-bg-url="">[18.08.2018: <a style="color: #ff0000;" href="https://sites.google.com/view/cibi-user-meeting-2018/home/registration">Registration</a> is now open!]</span></h2>
<p>Dear SeqAn followers and bioinformatics enthusiasts,</p>
<p>We are delighted to announce this yearâ€™s SeqAn User Meeting 2018 from 19<sup>thÂ </sup>to 21<sup>stÂ </sup>of September in Berlin. For the second time we will host it as a common User Meeting of the de.NBI Center for Integrative Bioinformatics (CIBI) and cordially invite researchers in the fields of genomics, proteomics, metabolomics, data mining and bioimage analysis to come together and share their knowledge.</p>
<p>We are excited to demonstrate for the first time the brand-new design of SeqAn3. In this context, we have a dedicated training day for beginners and advanced users on implementing applications with SeqAn3. With the help of your feedback and our experiences from past years we are convinced that SeqAn3 will make it genuinely easier to write applications with our library, while still providing the familiar performance of our algorithms and data structures.</p>
<p>Furthermore, we will have a training course for data scientists to learn how to process genomics data within KNIME using SeqAn and external NGS tools and the KNIME REST Web service functionality to query information from online web services such as ENSEMBL.</p>
<p>Find out more about the conference Â <a href="https://sites.google.com/view/cibi-user-meeting-2018/home">here</a> or register directly <a href="https://sites.google.com/view/cibi-user-meeting-2018/home/registration">here</a>.</p>
<p>Hope to see you soon!</p>
<p>Your SeqAn Team<div class="fusion-clearfix"></div></div></div></div></div></p>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn 3 @GCB 2018</title>
		<link>http://www.seqan.de/seqan_3_tutorial_at_the_gcb_2018/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Tue, 31 Jul 2018 12:14:14 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12783</guid>

					<description><![CDATA[We are happy to announce our tutorial,Â SeqAn3 â€“ a modern C++ library for efficient sequence analysis, in a full day tutorial at the GCB 2018 in Vienna. This is a beginner tutorial and focuses on tool programming using modern C++ and the SeqAn 3 library. SeqAn 3 is a major redesign of our sequence analysis [...]]]></description>
										<content:encoded><![CDATA[<p>We are happy to announce our tutorial,Â <strong>SeqAn3 â€“ a modern C++ library for efficient sequence analysis</strong>, in a full day tutorial at the GCB 2018 in Vienna.</p>
<p>This is a beginner tutorial and focuses on tool programming using modern C++ and the SeqAn 3 library. SeqAn 3 is a major redesign of our sequence analysis library based on modern C++ features such as Concepts and Ranges. You will learn all about these new cool features and how tool programming is simplified using SeqAn 3.</p>
<p>You can find more information regarding the tutorial <a href="https://gcb2018.csb.univie.ac.at/seqan3">at the GCB 2018</a>.</p>
<p>Please register for the tutorials directly at the GCB 2018:Â <a href="https://gcb2018.csb.univie.ac.at/registration">https://gcb2018.csb.univie.ac.at/registration</a></p>
<p>Please don&#8217;t hesitate writing us if you have more questions.</p>
<p>We are looking forward meeting you there.</p>
<p>p.s. If you can&#8217;t make it to the GCB 2018, we will present the brand new SeqAn 3 design atÂ our own User Meeting from September 19th &#8211; 21st in Berlin, together with a lot of show cases, programming and KNIME tutorials, and many great news for our users, regarding performance and efficiency. More information to our User Meeting will follow soon.</p>
<p>Also notify your colleagues about the opportunity to meet us at the <a href="http://www.seqan.de/seqan-programming-and-user-tutorial-at-the-eccb-2018/">EECB 2018</a>.</p>
<p>&nbsp;</p>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn @ECCB 2018</title>
		<link>http://www.seqan.de/seqan-programming-and-user-tutorial-at-the-eccb-2018/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Tue, 31 Jul 2018 12:05:06 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12779</guid>

					<description><![CDATA[We are happy to announce our tutorial,Â From an application to a fully integrated workflow â€“ Comprehensive software engineering with SeqAn, in a full day tutorial at the ECCB 2018 in Athen. This is a beginner tutorial and focuses on basic tool programming using C++ and the SeqAn library. In addition participants will learn how SeqAn [...]]]></description>
										<content:encoded><![CDATA[<p>We are happy to announce our tutorial,Â <strong>From an application to a fully integrated workflow â€“ Comprehensive software engineering with SeqAn</strong>, in a full day tutorial at the ECCB 2018 in Athen.</p>
<p>This is a beginner tutorial and focuses on basic tool programming using C++ and the SeqAn library. In addition participants will learn how SeqAn and external Â tools can be integrated into the Workflowsystem KNIME to build up comprehensive workflows connected to the strong capabilities of data mining included in KNIME.</p>
<p>You can find more information regarding the tutorial <a href="http://eccb18.org/tutorial-9/">at the ECCB</a>.</p>
<p>Please register for tutorial <strong>t9</strong>Â directly at the ECCB 2018:Â <a href="http://eccb18.org/registration/">http://eccb18.org/registration/</a></p>
<p>Please don&#8217;t hesitate writing us if you have more questions.</p>
<p>We are looking forward meeting you there.</p>
<p>p.s. If you can&#8217;t make it to ECCB 2018, we will present the brand new SeqAn 3 design at the <a href="https://gcb2018.csb.univie.ac.at/seqan3">GCB 2018</a>, and we will have our own User Meeting from September 19th &#8211; 21st, with a lot of show cases, programming and KNIME tutorials, and many great news for our users, regarding performance and efficiency. More information to our User Meeting will follow soon.</p>
<p>&nbsp;</p>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn User Meeting 2017</title>
		<link>http://www.seqan.de/seqan-ugm-2017/</link>
		
		<dc:creator><![CDATA[Rene Rahn]]></dc:creator>
		<pubDate>Fri, 14 Jul 2017 08:11:51 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://www.seqan.de/?p=12686</guid>

					<description><![CDATA[[01.08.2017] Registration now openÂ until 15th of August!!! Joint CIBI user meeting featuring: SeqAn:Â  C++ library of efficient algorithms and data structures for the analysis of biological sequence data. and OpenMS + MetFrag:Â High-Â­performance software for high-Â­throughput proteomics and metabolomics Dear friends and colleagues, we are happy to announce the 10thÂ CIBI User Meeting onÂ 25. September (09:00) until [...]]]></description>
										<content:encoded><![CDATA[<style>
table, th, td {
   border: 2px solid black;
   border-collapse: collapse;
}
th {
    text-align: center;
}
</style>
<h1 style="text-align: center;" data-fontsize="34" data-lineheight="48"><span style="color: #ff0000;"><strong><div class="fusion-fullwidth fullwidth-box fusion-builder-row-4 hundred-percent-fullwidth non-hundred-percent-height-scrolling"  style='background-color: rgba(255,255,255,0);background-position: center center;background-repeat: no-repeat;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px;'><div class="fusion-builder-row fusion-row "><div  class="fusion-layout-column fusion_builder_column fusion_builder_column_1_1 fusion-builder-column-3 fusion-one-full fusion-column-first fusion-column-last fusion-column-no-min-height 1_1"  style='margin-top:0px;margin-bottom:0px;'><div class="fusion-column-wrapper" style="background-position:left top;background-repeat:no-repeat;-webkit-background-size:cover;-moz-background-size:cover;-o-background-size:cover;background-size:cover;"   data-bg-url="">[01.08.2017] Registration now openÂ until 15th of August!!!</strong></span></h1>
<h1 style="text-align: center;" data-fontsize="34" data-lineheight="48"><strong>Joint CIBI user meeting featuring:</strong></h1>
<h2><strong>SeqAn:<em>Â  C++ library of efficient algorithms and data structures for the analysis of biological sequence data.</em></strong></h2>
<h2>and</h2>
<h2><strong>OpenMS + MetFrag:Â <em>High-Â­performance software for high-Â­throughput proteomics and metabolomics</em></strong></h2>
<h3 data-fontsize="20" data-lineheight="20">Dear friends and colleagues,</h3>
<p>we are happy to announce the 10<sup>th</sup>Â CIBI User Meeting onÂ 25. September (09:00) until 27. September 2017 (16:00)Â at Martin-Luther University, Steintor-Campus, Room SR7, Emil-Abderhalden-StraÃŸe, 06108 Halle, Germany</p>
<p><strong>Note: SeqAn, OpenMS, and MetFrag are developed by the Center for Integrative Bioinformatics (CIBI) and funded as a projectÂ of the german</strong><strong>Â network for bioinformatic infrastructure (de.NBI). Thus, for the first time, the SeqAnÂ userÂ </strong><strong>meeting is part of a larger CIBI user meeting. Participants may select if they want to participate in the SeqAn or OpenMS track. We also offer a dedicated track for users that want to develop computational tools with SeqAn and OpenMSÂ as well as an introduction to MetFrag.</strong></p>
<p>During the meeting, we will present the capabilities of CIBI tools SeqAn, OpenMS, and MetFrag and discuss your needs and future directions in sequence analysis and mass spectrometry.<br />
This yearâ€™s meeting will include an<strong>Â optional SeqAnÂ tutorial session on the first day (Monday, September 25)</strong>Â Â with theÂ <strong>general conference starting on Tuesday, September 26</strong>.</p>
<p>Traditionally, the schedule of our meetings is very flexible, allowing discussions and contributions from the participants.</p>
<p>You canÂ <strong>bring your data</strong>, aÂ <strong>wish list</strong>,Â <strong>ideas for collaboration projects</strong>, and alsoÂ <strong>give a talk about your project</strong>. This may include suggestions in areas where currently available software solutions were unsatisfactory. We would be very happy to welcome you in Halle this year.</p>
<h3 data-fontsize="20" data-lineheight="20">Accommodation</h3>
<p>Please take care of room reservations yourself.</p>
<p>We can suggest the following options. Prices etc. are only indicative and not guaranteed.</p>
<ul>
<li>Hotel Am Steintor (close to Workshop location)
<ul>
<li><a href="https://www.am-steintor.de/">https://www.am-steintor.de</a></li>
<li>40 EUR single room, 55 EUR double room</li>
</ul>
</li>
<li>Hostel No5 (20 min walk, or Tram connection, Nr 2 and Nr 5)
<ul>
<li><a href="http://www.hostel-no5.de/hostel_lage1.html">http://www.hostel-no5.de/hostel_lage1.html</a></li>
<li>single room: 25 EUR per night | double room: 40 EUR per night | triple room: 15 EUR per night and person</li>
<li>studio (single room, separate bath, kitchen): 40 EUR per night (single) or 50 EUR per night (double)</li>
<li>plus breakfast: 4 EUR per day and person (must be ordered before arrival)</li>
</ul>
</li>
<li>Hotel Ankerhof (20 min walk or direct Tram connection, Nr 2 and Nr 5)
<ul>
<li><a href="http://www.leipziggrafik.de/anker05/htdocs/anfahrt.shtml">http://www.leipziggrafik.de/anker05/htdocs/anfahrt.shtml</a></li>
<li>special rate (incl. breakfast) is 65 EUR per person and night with booking code: Â«Leibniz Institut fÃ¼r PflanzenbiochemieÂ»</li>
</ul>
</li>
<li>Further accommodation options:
<ul>
<li><a href="http://www.hrs.de/">http://www.hrs.de</a></li>
<li><a href="https://www.couchsurfing.org/">https://www.couchsurfing.org/</a></li>
</ul>
</li>
</ul>
<h3 data-fontsize="20" data-lineheight="20">Journey</h3>
<p><b>Directions to the Steintor Campus Uni Halle: </b><a href="https://goo.gl/ZNSmBM"><b>https://goo.gl/ZNSmBM</b></a></p>
<p><b>By train:</b> Halle is well connected within Germanyâ€™s network of IC and ICE trains <a href="https://reiseauskunft.bahn.de/bin/query2.exe/en">https://reiseauskunft.bahn.de/bin/query2.exe/en</a></p>
<p><b>By plane: </b>Nearest airports are Halle-Leipzig (LEJ), Erfurt (ERF), Berlin (TXL &amp; SXF).</p>
<p>Shortest connection is via Halle-Leipzig Airport, but flights might not run every day. Nice connections are possible via Berlin with train to Halle (Saale) and via Frankfurt (FRA) using the ICE train connection. Note that the Berlin airport BER is still not opened.</p>
<p><b>Public transport in Halle:</b> <a href="https://goo.gl/9Zo5vy">https://goo.gl/9Zo5vy</a>, <a href="http://www.havag.com/fahrplan/liniennetz-und-tarifzonenplan">http://www.havag.com/fahrplan/liniennetz-und-tarifzonenplan</a></p>
<h3 data-fontsize="20" data-lineheight="20"><strong>Registration</strong></h3>
<p>Participants need to register via <a href="https://goo.gl/forms/8WKW2T2OBrosfiuA3">https://goo.gl/forms/8WKW2T2OBrosfiuA3</a>. The deadline for the workshop registration is <strong>15</strong><b>. AugustÂ 2017</b>. Notification of acceptance is <strong>31</strong><b>. August 2017</b>. The workshop fee is 50Â â‚¬ for students (Undergrad / PhD), and 80Â â‚¬ for Industry, PostDoc, PI, Professor or above. Fees will cover coffee breaks, lunch, and the workshop dinner.</p>
<h3 data-fontsize="20" data-lineheight="20">Schedule</h3>
<table>
<thead>
<tr>
<th rowspan="2"></th>
<th colspan="3">Monday 2017-09-25</th>
<th>Tuesday 2017-09-26</th>
<th>Wednesday 2017-09-27</th>
</tr>
<tr>
<th>Room: SR7</th>
<th>Room: SR12</th>
<th>Room: SR8</th>
<th>Room: SR7</th>
<th>Room: SR7</th>
</tr>
</thead>
<tbody>
<tr>
<td>09:00 &#8211; 09:30</td>
<td colspan="3" align="center">Welcome and Introduction</td>
<td>Introduction to CIBI and its Tools<br />
Kohlbacher</p>
<p>What&#8217;s new in OpenMS<br />
Kohlbacher</td>
<td>Developer Talks</td>
</tr>
<tr>
<td>09:30 &#8211; 10:30</td>
<td>Workflow Tutorial<br />
with KNIME</td>
<td>Introduction to<br />
Programming with<br />
SeqAn</td>
<td>Introduction to<br />
Programming with<br />
OpenMS</td>
<td>What&#8217;s new in &#8230;</p>
<p>&#8230; SeqAn?<br />
&#8230; KNIME?<br />
&#8230; MetFrag?</td>
<td>Scientiffic talks</td>
</tr>
<tr>
<td>10:30 &#8211; 11:00</td>
<td colspan="3" align="center"><span style="color: #009999;">Coffee Break</span></td>
<td align="center"><span style="color: #009999;">Coffee Break</span></td>
<td align="center"><span style="color: #009999;">Coffee Break</span></td>
</tr>
<tr>
<td>11:00 &#8211; 12:00</td>
<td>Introduction to OpenMS<br />
and TOPP</td>
<td>Intro to MetFrag<br />
&amp; MassBank</td>
<td>SeqAn Tools and<br />
KNIME Workflows</td>
<td>Keynote: Ivo GroÃŸe<br />
Whatâ€™s New ?</td>
<td>Keynote: Wolfram Weckwerth<br />
System theoretical concepts<br />
in multidimensional genome-scale<br />
molecular data analysis</td>
</tr>
<tr>
<td>12:00 &#8211; 13:00</td>
<td colspan="3" align="center"><span style="color: #009999;">Lunch at <a href="https://www.studentenwerk-halle.de/hochschulgastronomie/halle/harzmensa/">Harzmensa</a> (700m walk)</span></td>
<td><span style="color: #009999;">Lunch at <a href="https://www.studentenwerk-halle.de/hochschulgastronomie/halle/harzmensa/">Harzmensa</a> (700m walk)</span></td>
<td><span style="color: #009999;">Lunch at <a href="https://www.studentenwerk-halle.de/hochschulgastronomie/halle/harzmensa/">Harzmensa</a> (700m walk)</span></td>
</tr>
<tr>
<td>13:00 &#8211; 14:30</td>
<td colspan="2">Advanced OpenMS</td>
<td>Advanced SeqAn Tools</td>
<td>Developer talks:<br />
SeqAn IPCC (Rahn)<br />
dynamic KNIME nodes (Pfeuffer)<br />
OpenMS talk</td>
<td>Scientiffic talks</td>
</tr>
<tr>
<td>14:30 &#8211; 15:00</td>
<td colspan="3" align="center"><span style="color: #009999;">Coffee Break</span></td>
<td align="center"><span style="color: #009999;">Coffee Break</span></td>
<td align="center"><span style="color: #009999;">Coffee Break</span></td>
</tr>
<tr>
<td>15:00 &#8211; 16:00</td>
<td>Advanced Proteomics and<br />
Metabolomics Workflows</td>
<td>Advanced NGS<br />
Data Analysis</td>
<td>BYOD/Hacking</td>
<td>Keynote: Greg Landrum<br />
t.b.a.</td>
<td>Scientiffic talks<br />
Wrap-up and departure</td>
</tr>
<tr>
<td>16:15 &#8211; 17:30</td>
<td>Integrated Multi-Omics<br />
Workflows</td>
<td>BYOD/Hacking</td>
<td>BYOD/Hacking</td>
<td>Scientiffic talks</td>
<td></td>
</tr>
<tr>
<td>17:30 &#8211; end</td>
<td colspan="3" align="center"><span style="color: #009999;">Self-paid DINNER at <a href="http://krugzumgruenenkranze.de/">Krug zum grÃ¼nen Kranze</a></span></td>
<td><span style="color: #009999;">Social event Tram City Tour<br />
and Workshop DINNER at<br />
<a href="http://www.halleschesbrauhaus.de/">Brauhaus Halle</a></span></td>
<td></td>
</tr>
</tbody>
</table>
<p>Speaker:<br />
We are excited to have Keynote talks by Wolfram Weckwerth (MOSYS, Uni Vienna), Greg Landrum (RDKIT/KNIME), Ivo Grosse (MLU Halle), and Steffen Neumann (IPB Halle)</p>
<h3 data-fontsize="20" data-lineheight="20">Contact</h3>
<p><a href="mailto:seqan-dev@lists.fu-berlin.de">SeqAn Team</a>Â â€“ scientific and program questions</p>
<p>See you in Halle!<br />
The CIBi Team<div class="fusion-clearfix"></div></div></div></div></div></p>
]]></content:encoded>
					
		
		
			</item>
	</channel>
</rss>
