<?xml version="1.0" encoding="UTF-8"?><rss version="2.0"
	xmlns:content="http://purl.org/rss/1.0/modules/content/"
	xmlns:wfw="http://wellformedweb.org/CommentAPI/"
	xmlns:dc="http://purl.org/dc/elements/1.1/"
	xmlns:atom="http://www.w3.org/2005/Atom"
	xmlns:sy="http://purl.org/rss/1.0/modules/syndication/"
	xmlns:slash="http://purl.org/rss/1.0/modules/slash/"
	>

<channel>
	<title>FUbln &#8211; Seqan</title>
	<atom:link href="https://www.seqan.de/author/fubln/feed/" rel="self" type="application/rss+xml" />
	<link>https://www.seqan.de</link>
	<description></description>
	<lastBuildDate>Mon, 27 Apr 2020 07:19:53 +0000</lastBuildDate>
	<language>en-US</language>
	<sy:updatePeriod>
	hourly	</sy:updatePeriod>
	<sy:updateFrequency>
	1	</sy:updateFrequency>
	<generator>https://wordpress.org/?v=5.6.1</generator>

<image>
	<url>../../../wp-content/uploads/2020/11/cropped-seqan_logo_4-2-32x32.png</url>
	<title>FUbln &#8211; Seqan</title>
	<link>https://www.seqan.de</link>
	<width>32</width>
	<height>32</height>
</image> 
	<item>
		<title>New IPCC</title>
		<link>https://www.seqan.de/ipcc/</link>
		
		<dc:creator><![CDATA[FUbln]]></dc:creator>
		<pubDate>Mon, 07 Dec 2015 13:25:48 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://46.101.193.202/?p=11657</guid>

					<description><![CDATA[The Intel® PCC program supports universities, institutions, and labs identified as leaders in their fields. It focuses on modernizing applications or software libraries to increase performance on modern microprocessors and coprocessors using parallel computing. Parallel computing involves the simultaneous operation of multiple processors running elements of a computer program at the same time and is [...]]]></description>
										<content:encoded><![CDATA[<p>The Intel® PCC program supports universities, institutions, and labs identified as leaders in their fields. It focuses on modernizing applications or software libraries to increase performance on modern microprocessors and coprocessors using parallel computing.</p>
<p>Parallel computing involves the simultaneous operation of multiple processors running elements of a computer program at the same time and is hence much faster than executing the operations sequentially, one after another. This necessitates routines to address the proper use of cores, caches, threads, and vector capabilities of the hardware as well as synchronization routines for parallel computations.</p>
<p>The latest <a href="https://software.intel.com/en-us/articles/intel-parallel-computing-center-at-freie-universitat-berlin">Intel® PCC</a> will be led by Knut Reinert, who is professor for Bioinformatics at the department of Mathematics and Computer Science at Freie Universität (FU) Berlin and a fellow at the Max-Planck-Institute for Molecular Genetics. Mr. Reinert’s research focuses on providing efficient tools for the analysis of Next Generation Sequencing (NGS) data stemming from a technology breakthrough several years ago which enables the cheap sequencing of terabytes of genomic sequence. The Reinert lab based its application development on well-designed algorithmic components and their implementation in the <a href="http://www.seqan.de/">SeqAn C++ software library</a>.</p>
<p>The SeqAn library is well-established and used worldwide in numerous analysis tools for NGS analysis. Since this year, it has also been supported in the CIBI center as part of the German bioinformatics infrastructure network (de.NBI), demonstrating its leading role in the development of new analysis tools for biomedical applications.</p>
<p><em>“We are very glad that Intel supports our vision for moving biomedical software development forward. SeqAn is a software library containing well designed key components for sequence analysis and we always thought that parallelizing and vectorizing key components of SeqAn will have a big impact on the field by accelerating many applications that make use of those components.”</em></p>
<p><strong>Knut Reinert, PI</strong></p>
<p>The center will work on abstracting primitives in SeqAn’s template based core to offer a unified interface to multicore and SIMD vector units including the Intel® Xeon and Intel® Xeon Phi<img src="https://s.w.org/images/core/emoji/13.0.1/72x72/2122.png" alt="™" class="wp-smiley" style="height: 1em; max-height: 1em;" /> coprocessors and then accelerate key routines like alignment algorithms or traversing data parallel containers. The generic design of SeqAn makes it well-suited for this approach.</p>
<p><em>&#8220;Intel regards SeqAn as a very promising software package that has all the right ingredients to considerably speed up Next Generation Sequencing analysis on modern Intel processors. We are looking forward to collaborating with Professor Reinert and his team to add our technical know-how about Intel® Architecture and combine it with his algorithmic expertise, and in this way turn SeqAn into a premier software tool in this domain of rapidly growing importance.”</em></p>
<p><strong>Kristina Kermanshahche</strong><br />
<strong> Chief Architect, Intel® Health &amp; Life Sciences</strong></p>
<p>The above strategy will accelerate existing and future applications based on the free SeqAn library (under BSD license) and make hardware acceleration easily available for developers. The FU Berlin will benefit from its role as an Intel® PCC by working with Intel experts and their software tools, and advanced technologies. The Reinert lab will incorporate its work in advanced tutorials and courses at FU Berlin and looks forward to sharing its findings at conferences such as the International Supercomputing Conference and the Intel® Xeon Phi<img src="https://s.w.org/images/core/emoji/13.0.1/72x72/2122.png" alt="™" class="wp-smiley" style="height: 1em; max-height: 1em;" /> Coprocessor User Group (IXPUG) meetings.</p>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn in de.NBI</title>
		<link>https://www.seqan.de/denbi/</link>
		
		<dc:creator><![CDATA[FUbln]]></dc:creator>
		<pubDate>Sat, 05 Dec 2015 08:19:05 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://46.101.193.202/?p=1</guid>

					<description><![CDATA[Das Bundeministerium für Bildung und Forschung (BMBF) fördert ab März für fünf Jahre das Deutsche Netzwerk für Bioinformatik Infrastruktur (de.NBI). Eines der acht Leistungszentren in diesem Netzwerk – das Zentrum für Integrative Bioinformatik (CIBi) – wird dabei für die nächsten fünf Jahre mit zwei Millionen Euro gefördert. CIBi ist ein gemeinsames Zentrum der Universitäten Tübingen [...]]]></description>
										<content:encoded><![CDATA[<p><strong>Das Bundeministerium für Bildung und Forschung (BMBF) fördert ab März für fünf Jahre das Deutsche Netzwerk für Bioinformatik Infrastruktur (de.NBI). Eines der acht Leistungszentren in diesem Netzwerk – das Zentrum für Integrative Bioinformatik (CIBi) – wird dabei für die nächsten fünf Jahre mit zwei Millionen Euro gefördert. CIBi ist ein gemeinsames Zentrum der Universitäten Tübingen und Konstanz sowie der Freien Universität Berlin.</strong></p>
<p>In der biomedizinischen Forschung hat die Einführung von neuen Sequenziermethoden und der hochauflösenden Massenspektrometrie einen Paradigmenwechsel ermöglicht. Die darauf basierenden Hochdurchsatzmethoden wie Genomik, Transkriptomik, Proteomik und Metabolomik – auch Omics-Methoden genannt – geben zwar sehr umfängliche und tiefe Einsichten in zelluläre Systeme, aber die erzeugten Daten sind äußerst umfangreich (im Bereich von Terabytes) und sehr komplex. Zunehmend werden heute auch Daten aus mehreren Technologien parallel erzeugt, zum Beispiel Daten zum Genom und zu den Proteinkonzentrationen in einer Zelle. Für Analyse und Interpretation solcher Datensätze sind daher innovative Algorithmen notwendig. Da ein einzelner Algorithmus für die Analyse dieser Daten nicht mehr ausreicht, werden diese Werkzeuge dann in komplexe Datenanalyse-Workflows eingebunden. Damit wird dann die automatisierte Auswertung selbst komplexester Daten möglich. Das BMBF fördert im Rahmen des Netzwerks die Weiterentwicklung von Algorithmen für die Analyse von Proteom- und Metabolomdaten entwickelt (Tübingen, Softwarepaket OpenMS, Prof. Oliver Kohlbacher), von Genom- und Transkriptomdaten (Berlin, Softwarepaket SeqAn, Prof. Knut Reinert) und der Integration dieser Tools in Workflows (Konstanz, Softwarepaket KNIME, Prof. Michael Berthold)</p>
<p>Das Zentrum für Integrative Bioinformatik ist eng an die anderen sieben Leistungszentren im Deutschen Netzwerk für Bioinformatik angebunden. Das Gesamtnetzwerk wird für fünf Jahre gefördert und nach drei Jahren zwischenevaluiert. Die Koordination des Netzwerks liegt bei der Universität Bielefeld</p>
]]></content:encoded>
					
		
		
			</item>
		<item>
		<title>SeqAn 2.0 released</title>
		<link>https://www.seqan.de/lorem-ipsum-2/</link>
		
		<dc:creator><![CDATA[FUbln]]></dc:creator>
		<pubDate>Wed, 17 Dec 2014 22:54:10 +0000</pubDate>
				<category><![CDATA[News]]></category>
		<guid isPermaLink="false">http://46.101.193.202/?p=12105</guid>

					<description><![CDATA[We are happy to announce the new release of SeqAn 2.0.0. We have many new features and applications for you and improved many parts in sense of usability, performance and stability. For example, we have improved the usability and the performance of the I/O-modules. Now BAM-I/O supports parallel read and write operations. We improved automatic [...]]]></description>
										<content:encoded><![CDATA[<p><strong>We are happy to announce the new release of SeqAn 2.0.0.</strong></p>
<p>We have many new features and applications for you and improved many parts in sense of usability, performance and stability.</p>
<p>For example, we have improved the usability and the performance of the I/O-modules. Now BAM-I/O supports<strong> parallel read and write</strong> operations. We improved automatic read/write operations for <strong>compressed file formats, like gzip, bzip2,</strong> etc..</p>
<p>We improved the performance of several data structures like the <strong>FMIndex</strong>, which runs now up to 4 times faster than the old version.</p>
<p>We extended the library with new features like the X-drop extensions for alignments allowing affine gap costs. We added a new realignment module as well as a translation module to translate amino acid alphabet into DNA alphabet.</p>
<p>We implemented many new apps like<strong> ANISE</strong> and <strong>BASIL</strong> for insert assembly, <strong>Fiona</strong> for read error correction, <strong>Yara,</strong> an enhanced read aligner replacing Masai, and many many more.</p>
<p>With SeqAn 2.0.0 we moved the complete sources to GitHub, which enhances the development cycle in a great way. We improved our build system and added <strong>Continous Integration</strong> builds with Travis CI. We also updated and improved the API documentation (<strong>docs.seqan.de</strong>) and switched the tutorials to <strong>seqan.readthedocs.org</strong>.</p>
<p>You can download the new release and the updated apps from www.seqan.de in the downloads section. You can get the complete sources of the SeqAn 2.0.0 from https://github/seqan/seqan as well.</p>
<p>Simply run:</p>
<p>git clone -b seqan-v2.0.0 https://github.com/seqan/seqan.git seqan-src</p>
<p>and start developing and having fun with SeqAn 2.0.0.</p>
<p>Enjoy!</p>
<p>The SeqAn Team</p>
<p>_______________________________________________<br />
seqan-dev mailing list<br />
seqan-dev@lists.fu-berlin.de<br />
https://lists.fu-berlin.de/listinfo/seqan-dev</p>
]]></content:encoded>
					
		
		
			</item>
	</channel>
</rss>
